# Abstract {-}

Exploratory data analysis  (EDA) is a vital element of the modern 
statistical workflow - it is an analyst’s first pass at 
understanding their data; revealing all it’s messes and (possibly) uncovering hidden insights. It is an iterative process that can 
lead to new hypotheses that can be tested and formalised using 
modelling. However, as data grows in complexity and there is an 
increase in the number of observations, and variables EDA becomes 
increasingly difficult.  This is certainly true in bioinformatic 
data analysis, where there are constraints computation time and 
memory, as well as a lack of guidelines or tooling for performing 
EDA.

This thesis presents a set of tools for performing exploratory 
data analysis in the context of genomics and transcriptomics.  The first tool, “plryanges” provides a domain specific language for 
reasoning about range based genomics data that simplifies common 
operations such as overlaps and aggregations.  Next I extend 
“plyranges”  to develop “superintronic” which presents an 
approach for discovering interesting genomic regions within an 
experimental design. We apply this tool to explore intron signal 
in transcriptomic data. Finally, I turn my attention to focus on 
embedding methods currently used for visualisation of large data 
sets… 

\newpage
