# Exploratory coverage analysis with superintronic and plyranges {#ch:intron}

Here we consider a tidy-data approach for exploring
estimated coverage from RNA-seq data. We establish a simple 
framework, for aggregating across experimental design, and annotated genomic regions to discover 'interesting' coverage trace plots. We highlight how this framework can be used to
develop data descriptions that find putative genes with
intron retention. Our framework is implemented in a software
package called **superintronic**, available at https://github.com/sa-lee/superintronic.


## Introduction

In high-throughput sequencing data sets, coverage is the estimated number of
reads that overlap a single position of the reference genome, and is important for assessing sequencing data quality and used in many different aspects of omics analysis such peak-calling in ChIP-seq or variant calling in DNA-seq [@Sims2014-qk]. Here we emphasise looking at coverage traces to find biological events of interest, rather than only relying on numerical summaries of the data. By faceting these traces over combinations of the experimental design and with their biological context such as gene annotations, we can gain an
insight into biological signal under study. 
Of course, due to the sheer size of most reference genome annotations 
it would take an extremely long time for an analyst to look at every 
single possible region where there is some interesting signal within the
coverage trace. Because visualisation does not scale, we need to search the possible space of coverage traces and provide diagnostics for
identifying traces with interesting biology. We have taken inspiration from
visualisation literature; in particular the idea of scatter plot diagnostics ("scagnostics") for summarising the space of all possible 2D scatter plots
to a small number of descriptors of each scatter plots properties such as 
density or monotonicity [@Friedman2002-yx;@Wilkinson2005-xt].  Similarly,
searching for unusual time series via estimating descriptors such as seasonality or autocorrelation, and visualising those descriptors instead [@Hyndman2015-sc].

Although there are many flexible and powerful software tools, like BEDtools or deepTools, for exploring and estimating coverage from common genomic data formats [@Quinlan2010-gc; @Ramirez2014-sh], it is advantageous to have tooling
that is tightly coupled to a statistical computing language such as R. This allows interoperability between other software packages for data wrangling, visualisation and modelling within the ecosystem of the language that may not
be possible with a single command line tool. There are also gains in reproducibility as analysts do not have to move between multiple software suites to explore their processed transcriptomics data. 

In @Lee2019-mf, we showed that there is evidence that a major
source of intron reads in RNA-seq datasets is pre-mRNA, and sought data analysis techniques to unravel different aspects of intron signal. In light of this,
we made the assumption that most intron reads do not necessarily point to
intron retention (IR) events, and developed a workflow based on combining
multiple summary statistics, "data descriptors", to find coverage traces that appear to have IR-like events by collapsing coverage scores over a design matrix alongside the exonic and intronic parts of a gene. To do this we developed a new R package called **superintronic** that provides tooling for exploratory 
coverage analysis by extending and integrating our previous software package,
**plyranges** [@Lee2019]. 

In this chapter, we describe methodology for establishing data descriptors by turning coverage vectors into long form tidy data using **superintronic**
and **plyranges**. We provide a workflow below
using a zebrafish RNA-seq dataset for developing data descriptors to find IR like coverage traces within genes known to have minor class splicing events. 


## Methods

**superintronic** is an R package used for estimating, representing,
and visualising per-base coverage scores, that can then be 
flexibly summarised over factors within an experimental design and collapsed
over regions of the genome using our companion package **plyranges**. The
aspects of this combined workflow are summarised in figure \@ref(fig:si-fig). 

(ref:si-fig) An overview of the **superintronic** and **plyranges** workflow. Coverage is estimated directly using the design matrix that contains a source column pointing to the locations of BAM files. The long-form representation is output as a **GRanges** object, and contains columns that were part of the design. Additional annotations are add with join functions, here we show the particular case of expanding the coverage **GRanges** to include exonic and intronic parts of a gene. This object can be further analysed using **plyranges** and our data descriptors approach, and then descriptors can be visualised as a scatter plot matrix with the `GGally` and `ggplot2` packages [@ggally; @Wickham2016-gz]. Coverage traces can be directly generated with **superintronic** and collapsed over parts of the design matrix to identify differences between groups.

```{r si-fig, fig.cap = "(ref:si-fig)", echo = FALSE}
knitr::include_graphics(here::here("img", "si-workflow.png"))
```

### Representation of coverage estimation

The per base coverage score is estimated directly from one or more BAM files
that represent the units within the experimental design,
along with an optional experimental design table that is returns 
a long-form tidy _GRanges_ data structure. The coverage estimation is
computed via the **Rsamtools** package, and
users have the ability to estimate coverage in parallel and drop regions
in the genome where there is no coverage [@rsamtools; @granges]. This
representation is tidy, since each row of the resulting _GRanges_ data
structure corresponds to position(s) within a given sample with a given 
coverage score alongside any variables such as biological group. While the long-form representation repeats the same information
for a sample within the design, the size of the resulting _GRanges_ in
memory can be compressed using run-length encoding for any categorical 
variable. This representation allows us to easily transform the coverage scores
and integrate annotations using the **plyranges** grammar, and visualise traces using **ggplot2** [@Wickham2016-gz].

### Integration of external annotations

External reference annotations, perhaps transcripts or exons, 
can be coerced to _GRanges_ objects, are  incorporated into the coverage _GRanges_ by taking the intersection of
the annotation with the _GRanges_ using an overlap intersect join from the **plyranges** software. The resulting intersection will now contain the
per base coverage that are overlapped the genomic features in the annotation,
along side any metadata about the features themselves. Since our main workflow
interest is in discovering coverage traces with IR profiles, **superintronic**
provides some syntactic sugar for unravelling gene annotations into their exonic
and intronic parts, and intersecting them with a coverage _GRanges_.

### Discovery of regions of interest via 'data descriptors'

Once the coverage _GRanges_ has reference genomic features, data 
descriptors can be computed via collecting summary statistics across
the factors of the experimental design and features of interest. This can be
achieved using **plyranges** directly by first grouping across variables
of interest, computing descriptors defined by **superintronic** and then
pivoting the results into a wide form table for additional processing or
visualisation. There are many descriptors defined by **superintronic**
that are weighted statistics (as we have to account for the number of bases
covered, or the width of the range) of the coverage score, such as the mean 
and standard deviation. There are also descriptors that can be used to find
the number of times the coverage trace is above a certain number of bases
or score. To find coverage traces that have unusual descriptors, 
the descriptors can be visualised directly as a scatter plot matrix. After that
thresholds can be applied to filter the genomic features that had extreme
descriptors on the coverage _GRanges_, and the traces can be visualised. 
By default **superintronic** displays coverage traces oriented from the 5' to 3' end of the gene, with the `view_coverage()` function. Traces can be highlighted according to a genomic feature of interest, figures in this chapter
have orange areas corresponding to intronic parts 
of the gene, while dark green areas referring to exonic parts.

## A workflow for uncovering intron retention in  a zebrafish experiment

In the study of gene regulation, there is much interest in uncovering the
effects of aberrant minor class splicing (called U12 splicing) on the transcriptome.
Minor class splicing is a regulatory process where a class of 
introns (called U12 introns in this case but there exist other classes in eukaryotic organisms such as U2 introns) are removed from pre-mRNA prior to gene expression [@Turunen2013-qu]. The removal of these introns is catalysed by small ribonucleoproteins (snRNPs)
which identify key motifs and branch sites in the intron to begin splicing [@Markmiller2014-kg].
Here we describe an exploratory workflow for finding coverage traces
with evidence for intron retention using RNA-seq data from a knockout experiment in zebrafish obtained from the Heath Lab at Walter and Eliza Hall Institute. Code
for this analysis is available at https://github.com/sa-lee/thesis/tree/master/scripts/superintronic.R.

The data consist of 11 zebrafish samples from single-end polyA enriched RNA-seq libraries pooled from zebrafish larvae. The experimental factors looked at combinations of genotype (whether the gene rnpc3 has been knocked out or not) and line (whether the zebrafish larvae have the caliban *cal* or mutant caliban *zm-cal* phenotype). Within each combination there are three biological replicates, except for the combination *zm-cal* and wild-type rnpc3 which
had two replicates.

FASTQ files were aligned to the GRCz11 reference genome using subjunc with
the default parameters called from Rsubread to produce BAM files for each
sample [@Liao2013-eb; @Liao2019-qn]. The coverage was then estimated directly from the set of BAM files
using **superintronic** into the long form _GRanges_ representation we described above. 

The gene annotation files were
obtained as GFF files from RefSeq and used that to construct the exonic
and intronic parts of each gene as _GRanges_ object using *superintronic*. 
We further filtered genes that had a single exon or genes that were overlapping
others in the annotation and that were not on the main contigs of the reference genome (i.e. excluding mitochondria) to simplify our analysis and reduce any coverage ambiguity. This left 18,270 genes available for 
computing data descriptors on in order to detecting IR-like coverage traces. Across each combination of genotype and line, we first $log_2$ transformed the coverage score with an offset of one, and then intersected the coverage _GRanges_ with exonic and intronic features of each gene. For each gene,
the weighted mean and standard deviation of the log-transformed coverage score were computed over all intron and exon parts within the groupings of 
genotype and line. We also computed the bases above descriptor for intron
features. It refers to the total number of bases within an intron that has a score above the overall average exon coverage score. The scatter plot 
matrix view of these descriptors is shown in figure \@ref(fig:si-splom)
for a single biological group in the experiment. Using these views and
by summarising over the data descriptors, we came up with thresholds
for finding genes that have IR-like traces within each biological group.

(ref:si-splom) A hexbin scatter plot matrix of the data descriptors estimated for the *cal* rnpc3 knock out zebrafish line. To identify coverage traces with IR like events, we want a set of descriptors that will find genes with the following characteristics: the gene is "expressed" that has a large number of intron bases relative to the coverage of other intron features, and has  relatively stable coverage within exon features. To do this we looked at the  descriptor, as well as computing the the mean and standard deviation of both exon and intron features.

```{r si-splom, fig.cap = "(ref:si-splom)", out.width="80%"}
knitr::include_graphics(here::here("img", "Cal-Rnpc3_KO-pairs-plot.png"))
```

We selected genes with the following thresholds:  genes have an average exon log-coverage greater than the mean of average exon log-coverage values across all genes, have an average intron log-coverage greater than the average standard deviation of exon log-coverage values over all genes, and
the standard deviation of intron log-coverage values is twice the standard
deviation of exon log-coverage values. That is, we are selecting genes that
are expressed but  have large average intron expression that is more variable than the gene's exon expression. This results in a total of 86 genes selected
to link back to their underlying coverage traces, with the overlaps shown
in the UpSet plot in figure \@ref(fig:si-upset) [@Lex2014-bh]. The procedure produces gene coverage traces with known minor class splice sites affected by the knockout procedure such as ccdc43 and nat15 (figures \@ref(fig:si-cvg-ccdc43) and \@ref(fig:si-cvg-nat15)), as well as some that appear to have U12 intron retention like events such as mapk3 or tspan31 (figures \@ref(fig:si-cvg-mapk3) and \@ref(fig:si-cvg-tspan31)) that affect other parts of the gene.

(ref:si-upset) Gene overlaps found between each combination of genotype and line using the thresholds defined in the text. Our procedure mostly finds genes with IR like profiles across all groups (28 shared between all four) or that is unique to a single group, since we do not consider looking at differential IR and run our thresholding separately for each group of replicates. The rnpc3 knockout lines 
share the largest overlap in results. 

```{r si-upset, echo=FALSE, fig.cap = "(ref:si-upset)", out.width="80%"}
knitr::include_graphics(here::here("img", "si-olaps.png"))
```

(ref:si-cvg-ccdc43) The ccdc43 gene is known to have enhanced U12 intron retention in the caliban phenotype and increase retention when rnpc3 is knocked out as can be seen directly from the intron located at around 31.603Mb.  

```{r si-cvg-ccdc43, echo=FALSE, fig.cap = "(ref:si-cvg-ccdc43)", out.width="80%"}
knitr::include_graphics(here::here("img", "ccdc43.pdf"))
```

(ref:si-cvg-nat15) The gene nat15 exhibits another example of U12 intron retention (located at around 27.93Mb) in the rnpc3 knockout groups.  

```{r si-cvg-nat15,  echo=FALSE, fig.cap = "(ref:si-cvg-nat15)", out.width="80%"}
knitr::include_graphics(here::here("img", "nat15.pdf"))
```

(ref:si-cvg-mapk3) The gene mapk3 appears to have intron retention close to the start of the gene that have different impacts downstream, which result in different IR profiles between the knockout and wild-type groups.  

```{r si-cvg-mapk3,  echo=FALSE, fig.cap = "(ref:si-cvg-mapk3)", out.width="80%"}
knitr::include_graphics(here::here("img", "mapk3.pdf"))
```

(ref:si-cvg-tspan31) The gene tspan31 has a potential cryptic splice site within the intron retained in the knockout groups.

```{r si-cvg-tspan31, echo=FALSE, fig.cap = "(ref:si-cvg-tspan31)", out.width="80%"}
knitr::include_graphics(here::here("img", "tspan31.pdf"))
```

## Discussion

We have shown how to coverage can be represented in the tidy data framework
and integrated with experimental metadata and reference annotations. This
framework allowed us to build data descriptions that are simple aggregations of
various aspects of genomic features over factors within a designed experiment
and link those descriptions to their underlying coverage traces.

Our zebrafish workflow shows that our approach using **superintronic** 
and **plyranges** is able to uncover interesting biological signals in a
purely data-driven manner. We did not include additional information
that could have been useful when deriving our selected genes, such as
sequence motifs for U12 class of introns, or exploit the experimental
design to find differential IR like profiles. However, if that was of interest,
one could look at the overlaps, like we did in figure \@ref(fig:si-upset),
or combine our data descriptors with external estimates using **limma** 
[@Ritchie2015-mj], like our proposed index method in @Lee2019-mf. The gene
candiates obtain by our thresholds have been validated by the Heath lab
using qPCR. 

Although the example we have explored has related to finding coverage
traces with IR-like events, the workflow of building and then visualising
data descriptors could be generalised to other types of omics analyses,
such as peaking finding in ChIP-seq, and to use more sophisticated methods for identifying  thresholds of 'interesting' traces. Our approach would also greatly benefit from interactive graphics that dynamically link say a gene description to
an underlying coverage trace, for rapid exploration. This is left for 
future work.

## Acknowledgements {-}

We would like to thank Dr Alexandra Garnham, Dr Stephen Mieruszynski and 
Associate Professor Joan Heath
for providing the zebrafish data and helping interpret the results from the workflow.
