<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.1 Introduction | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="3.1 Introduction | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="3.1 Introduction | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.1 Introduction | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="3-ch-fluentGenomics.html"/>
<link rel="next" href="3-2-se.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-acknowledgements-1.html"><a href="2-5-acknowledgements-1.html"><i class="fa fa-check"></i><b>2.5</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.6" data-path="2-6-availability-of-data-and-materials.html"><a href="2-6-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.6</b> Availability of Data and Materials</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="3.7" data-path="3-7-acknowledgements-2.html"><a href="3-7-acknowledgements-2.html"><i class="fa fa-check"></i><b>3.7</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Tidy coverage analysis uncovers intron retention</a></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Spurious embeddings from non-linear dimensionality reduction methods and how to find them</a></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion and future plans</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p>In this workflow, we examine a subset of the RNA-seq and ATAC-seq data from
<span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span>, a study that involved treatment of macrophage cell lines from a number
of human donors with interferon gamma (IFNg), <em>Salmonella</em> infection, or both
treatments combined. <span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span> examined gene expression and chromatin
accessibility in a subset of 86 successfully differentiated induced pluripotent
stem cells (iPSC) lines, and compared baseline and response with respect to
chromatin accessibility and gene expression at specific quantitative trait loci
(QTL). The authors found that many of the stimulus-specific expression QTL were
already detectable as chromatin QTL in naive cells, and further hypothesize
about the nature and role of transcription factors implicated in the response
to stimulus.</p>
<p>We will perform a much simpler analysis than the one found in <span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span>, using
their publicly available RNA-seq and ATAC-seq data (ignoring the genotypes). We
will examine the effect of IFNg stimulation on gene expression and chromatin
accessibility, and look to see if there is an enrichment of differentially
accessible (DA) ATAC-seq peaks in the vicinity of differentially expressed (DE)
genes. This is plausible, as the transcriptomic response to IFNg stimulation
may be mediated through binding of regulatory proteins to accessible regions,
and this binding may increase the accessibility of those regions such that it
can be detected by ATAC-seq.</p>
<div class="figure" style="text-align: center"><span id="fig:workflow"></span>
<img src="img/workflow.png" alt="An overview of the fluent genomics workflow. First, we import data as a SummarizedExperiment object, which enables interoperability with downstream analysis packages. Then we model our assay data, using the existing Bioconductor packages DESeq2 and limma. We take the results of our models for each assay with respect to their genomic coordinates, and integrate them. First, we compute the overlap between the results of each assay, then aggregate over the combined genomic regions, and finally summarize to compare enrichment for differentially expressed genes to non differentially expressed genes. The final output can be used for downstream visualization or further transformation." width="100%" />
<p class="caption">
Figure 1.6: An overview of the fluent genomics workflow. First, we <em>import</em> data as a <em>SummarizedExperiment</em> object, which enables interoperability with downstream analysis packages. Then we <em>model</em> our assay data, using the existing Bioconductor packages <code>DESeq2</code> and <code>limma</code>. We take the results of our models for each assay with respect to their genomic coordinates, and <em>integrate</em> them. First, we compute the overlap between the results of each assay, then aggregate over the combined genomic regions, and finally summarize to compare enrichment for differentially expressed genes to non differentially expressed genes. The final output can be used for downstream visualization or further transformation.
</p>
</div>

<p>Throughout the workflow (Figure <a href="3-1-introduction.html#fig:workflow">1.6</a>), we will use existing
Bioconductor infrastructure to understand these datasets. In particular, we
will emphasize the use of the Bioconductor packages <code>plyranges</code> and <code>tximeta</code>.
The <code>plyranges</code> package fluently transforms data tied to genomic ranges using
operations like shifting, window construction, overlap detection, etc. It is
described by <span class="citation">Lee, Cook, and Lawrence (<a href="bibliography.html#ref-Lee2019" role="doc-biblioref">2019</a>)</span> and leverages underlying core Bioconductor infrastructure
<span class="citation">(Lawrence et al. <a href="bibliography.html#ref-granges" role="doc-biblioref">2013</a><a href="bibliography.html#ref-granges" role="doc-biblioref">b</a>; Huber et al. <a href="bibliography.html#ref-bioc" role="doc-biblioref">2015</a><a href="bibliography.html#ref-bioc" role="doc-biblioref">b</a>)</span> and the <code>tidyverse</code> design principles <span class="citation">Wickham et al. (<a href="bibliography.html#ref-tidyverse" role="doc-biblioref">2019</a>)</span>.</p>
<p>The <code>tximeta</code> package described by <span class="citation">Love et al. (<a href="bibliography.html#ref-Love2019-tximeta" role="doc-biblioref">2019</a>)</span> is used to read RNA-seq
quantification data into R/Bioconductor, such that the transcript ranges and
their provenance are automatically attached to the object containing expression
values and differential expression results.</p>
<div id="experimental-data" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Experimental Data</h3>
<p>The data used in this workflow is available from two packages: the <code>macrophage</code>
Bioconductor ExperimentData package and from the workflow package
<code>fluentGenomics</code>.</p>
<p>The <code>macrophage</code> package contains RNA-seq quantification from 24 RNA-seq
samples, a subset of the RNA-seq samples generated and analyzed by <span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span>. The
paired-end reads were quantified using <em>Salmon</em> <span class="citation">(Patro et al. <a href="bibliography.html#ref-salmon" role="doc-biblioref">2017</a>)</span>, using the Gencode 29
human reference transcripts <span class="citation">(Frankish, GENCODE-consoritum, and Flicek <a href="bibliography.html#ref-gencode" role="doc-biblioref">2018</a>)</span>. For more details on quantification, and
the exact code used, consult the vignette of the
<a href="http://bioconductor.org/packages/macrophage">macrophage</a> package. The package
also contains the <code>Snakemake</code> file that was used to distribute the <em>Salmon</em>
quantification jobs on a cluster <span class="citation">(KÃ¶ster and Rahmann <a href="bibliography.html#ref-snakemake" role="doc-biblioref">2012</a>)</span>.</p>
<p>The <code>fluentGenomics</code> package contains functionality to download and generate a
cached <em>SummarizedExperiment</em> object from the normalized ATAC-seq data provided
by <span class="citation">Alasoo and Gaffney (<a href="bibliography.html#ref-alasooZenodo" role="doc-biblioref">2017</a>)</span>. This object contains all 145 ATAC-seq samples across all
experimental conditions as analyzed by <span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span>. The data can be also be
downloaded directly from the
<a href="https://zenodo.org/record/1188300#.XIAhXlNKjOQ">Zenodo</a> deposition.</p>
<p>The following code loads the path to the cached data file, or if it is not
present, will create the cache and generate a <em>SummarizedExperiment</em> using the
the <code>BiocFileCache</code> package <span class="citation">(Shepherd and Morgan <a href="bibliography.html#ref-bcfilecache" role="doc-biblioref">2019</a>)</span>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"><span class="kw">library</span>(fluentGenomics)</a>
<a class="sourceLine" id="cb15-2" title="2">path_to_se &lt;-<span class="st"> </span><span class="kw">cache_atac_se</span>()</a></code></pre></div>
<p>We can then read the cached file and assign it to an object called <code>atac</code>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">atac &lt;-<span class="st"> </span><span class="kw">readRDS</span>(path_to_se)</a></code></pre></div>
<p>A precise description of how we obtained this <em>SummarizedExperiment</em> object can be found in section <a href="3-2-se.html#atac">3.2.2</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="3-ch-fluentGenomics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="3-2-se.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//03-fluentGenomics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
