<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.3 Model assays | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="3.3 Model assays | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="3.3 Model assays | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.3 Model assays | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="3-2-se.html"/>
<link rel="next" href="3-4-integrate-ranges.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-sec-tidygranges.html"><a href="1-1-sec-tidygranges.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-sec-integrate.html"><a href="1-2-sec-integrate.html"><i class="fa fa-check"></i><b>1.2</b> Integration of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-sec-coverage.html"><a href="1-3-sec-coverage.html"><i class="fa fa-check"></i><b>1.3</b> Representation of genomic data structures</a></li>
<li class="chapter" data-level="1.4" data-path="1-4-sec-va.html"><a href="1-4-sec-va.html"><i class="fa fa-check"></i><b>1.4</b> Interactive visualisation for high-dimensional data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <strong>plyranges</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <strong>plyranges</strong> and <strong>tximeta</strong></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <strong>tximeta</strong> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <strong>plyranges</strong></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Exploratory coverage analysis with superintronic and plyranges</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.2.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.2.2" data-path="4-2-methods.html"><a href="4-2-methods.html#integration-of-external-annotations"><i class="fa fa-check"></i><b>4.2.2</b> Integration of external annotations</a></li>
<li class="chapter" data-level="4.2.3" data-path="4-2-methods.html"><a href="4-2-methods.html#discovery-of-regions-of-interest-via-data-descriptors"><i class="fa fa-check"></i><b>4.2.3</b> Discovery of regions of interest via ‘data descriptors’</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><i class="fa fa-check"></i><b>4.3</b> A workflow for uncovering intron retention in a zebrafish experiment</a></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-3.html"><a href="acknowledgements-3.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-overview-of-dimension-reduction.html"><a href="5-2-overview-of-dimension-reduction.html"><i class="fa fa-check"></i><b>5.2</b> Overview of Dimension Reduction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="5-2-overview-of-dimension-reduction.html"><a href="5-2-overview-of-dimension-reduction.html#tours-explore-the-subspace-of-d-dimensional-projections"><i class="fa fa-check"></i><b>5.2.1</b> Tours explore the subspace of <span class="math inline">\(d\)</span>-dimensional projections</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html"><i class="fa fa-check"></i><b>5.3</b> Visual Design</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.3.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.3.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.3.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.3.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html"><i class="fa fa-check"></i><b>5.4</b> Software Infrastructure</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.4.1</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#linking-and-highlighting-views-via-interactions"><i class="fa fa-check"></i><b>5.4.2</b> Linking and highlighting views via interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html"><i class="fa fa-check"></i><b>5.5</b> Case Studies</a><ul>
<li class="chapter" data-level="5.5.1" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-1-exploring-spherical-gaussian-clusters"><i class="fa fa-check"></i><b>5.5.1</b> Case Study 1: Exploring spherical Gaussian clusters</a></li>
<li class="chapter" data-level="5.5.2" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-2-exploring-spherical-gaussian-clusters-with-hierarchical-structure"><i class="fa fa-check"></i><b>5.5.2</b> Case Study 2: Exploring spherical Gaussian clusters with hierarchical structure</a></li>
<li class="chapter" data-level="5.5.3" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-3-exploring-data-with-piecewise-linear-structure"><i class="fa fa-check"></i><b>5.5.3</b> Case Study 3: Exploring data with piecewise linear structure</a></li>
<li class="chapter" data-level="5.5.4" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-4-clustering-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.5.4</b> Case Study 4: Clustering single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="5-6-discussion-3.html"><a href="5-6-discussion-3.html"><i class="fa fa-check"></i><b>5.6</b> Discussion</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-4.html"><a href="acknowledgements-4.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="supplementary-materials.html"><a href="supplementary-materials.html"><i class="fa fa-check"></i>Supplementary Materials</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-software-development.html"><a href="6-1-software-development.html"><i class="fa fa-check"></i><b>6.1</b> Software Development</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-further-work.html"><a href="6-2-further-work.html"><i class="fa fa-check"></i><b>6.2</b> Further Work</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="A-getting-started-with-the-plyranges-package.html"><a href="A-getting-started-with-the-plyranges-package.html"><i class="fa fa-check"></i><b>A</b> Getting started with the <strong>plyranges</strong> package</a><ul>
<li class="chapter" data-level="A.1" data-path="A-1-ranges-revisited.html"><a href="A-1-ranges-revisited.html"><i class="fa fa-check"></i><b>A.1</b> <em>Ranges</em> revisited</a></li>
<li class="chapter" data-level="A.2" data-path="A-2-constructing-ranges.html"><a href="A-2-constructing-ranges.html"><i class="fa fa-check"></i><b>A.2</b> Constructing <em>Ranges</em></a></li>
<li class="chapter" data-level="A.3" data-path="A-3-arithmetic-on-ranges.html"><a href="A-3-arithmetic-on-ranges.html"><i class="fa fa-check"></i><b>A.3</b> Arithmetic on Ranges</a></li>
<li class="chapter" data-level="A.4" data-path="A-4-grouping-ranges.html"><a href="A-4-grouping-ranges.html"><i class="fa fa-check"></i><b>A.4</b> Grouping <em>Ranges</em></a></li>
<li class="chapter" data-level="A.5" data-path="A-5-restricting-ranges.html"><a href="A-5-restricting-ranges.html"><i class="fa fa-check"></i><b>A.5</b> Restricting <em>Ranges</em></a></li>
<li class="chapter" data-level="A.6" data-path="A-6-summarising-ranges.html"><a href="A-6-summarising-ranges.html"><i class="fa fa-check"></i><b>A.6</b> Summarising <em>Ranges</em></a></li>
<li class="chapter" data-level="A.7" data-path="A-7-joins-or-another-way-at-looking-at-overlaps-between-ranges.html"><a href="A-7-joins-or-another-way-at-looking-at-overlaps-between-ranges.html"><i class="fa fa-check"></i><b>A.7</b> Joins, or another way at looking at overlaps between <em>Ranges</em></a></li>
<li class="chapter" data-level="A.8" data-path="A-8-grouping-by-overlaps.html"><a href="A-8-grouping-by-overlaps.html"><i class="fa fa-check"></i><b>A.8</b> Grouping by overlaps</a></li>
<li class="chapter" data-level="A.9" data-path="A-9-reading-genomic-files.html"><a href="A-9-reading-genomic-files.html"><i class="fa fa-check"></i><b>A.9</b> Reading Genomic Files</a></li>
<li class="chapter" data-level="A.10" data-path="A-10-learning-more.html"><a href="A-10-learning-more.html"><i class="fa fa-check"></i><b>A.10</b> Learning more</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="model-assays" class="section level2">
<h2><span class="header-section-number">3.3</span> Model assays</h2>
<div id="rna-seq-differential-gene-expression-analysis" class="section level3">
<h3><span class="header-section-number">3.3.1</span> RNA-seq differential gene expression analysis</h3>
<p>We can easily run a differential expression analysis with <strong>DESeq2</strong> using the
following code chunks <span class="citation">(Love, Huber, and Anders <a href="bibliography.html#ref-Love2014" role="doc-biblioref">2014</a>)</span>. The design formula indicates that we want to
control for the donor baselines (<em>line</em>) and test for differences in gene
expression on the condition. For a more comprehensive discussion of DE
workflows in Bioconductor see <span class="citation">Love et al. (<a href="bibliography.html#ref-Love2016-f1000" role="doc-biblioref">2016</a>)</span> and <span class="citation">Law et al. (<a href="bibliography.html#ref-Law2018-f1000" role="doc-biblioref">2018</a>)</span>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="3-3-model-assays.html#cb29-1"></a><span class="kw">library</span>(DESeq2)</span>
<span id="cb29-2"><a href="3-3-model-assays.html#cb29-2"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSet</span>(gse, <span class="op">~</span>line <span class="op">+</span><span class="st"> </span>condition)</span>
<span id="cb29-3"><a href="3-3-model-assays.html#cb29-3"></a><span class="co"># filter out lowly expressed genes</span></span>
<span id="cb29-4"><a href="3-3-model-assays.html#cb29-4"></a><span class="co"># at least 10 counts in at least 6 samples</span></span>
<span id="cb29-5"><a href="3-3-model-assays.html#cb29-5"></a>keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(dds) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">10</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">6</span></span>
<span id="cb29-6"><a href="3-3-model-assays.html#cb29-6"></a>dds &lt;-<span class="st"> </span>dds[keep,]</span></code></pre></div>
<p>The model is fit with the following line of code:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="3-3-model-assays.html#cb30-1"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</span></code></pre></div>
<p>Below we set the contrast on the condition variable, indicating we are
estimating the <span class="math inline">\(\log_2\)</span> fold change (LFC) of IFNg stimulated cell lines against
naive cell lines. We are interested in LFC greater than 1 at a nominal false
discovery rate (FDR) of 1%.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="3-3-model-assays.html#cb31-1"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds,</span>
<span id="cb31-2"><a href="3-3-model-assays.html#cb31-2"></a>               <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;IFNg&quot;</span>,<span class="st">&quot;naive&quot;</span>),</span>
<span id="cb31-3"><a href="3-3-model-assays.html#cb31-3"></a>               <span class="dt">lfcThreshold=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.01</span>)</span></code></pre></div>
<p>The <code>results()</code> function extracts a summary of the DE analysis: in this case for each gene we have the LFC comparing the two cell lines, the Wald test statistic of LFC from the fitted negative binomial GLM, and associated p-value and corrected p-value accounting for the FDR. To see the results of the expression analysis, we can generate a summary table and a mean-abundance (MA) plot <span class="citation">(Dudoit et al. <a href="bibliography.html#ref-Dudoit2002-vz" role="doc-biblioref">2002</a>)</span>:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="3-3-model-assays.html#cb32-1"></a><span class="kw">summary</span>(res)</span></code></pre></div>
<pre><code>#&gt; 
#&gt; out of 17806 with nonzero total read count
#&gt; adjusted p-value &lt; 0.01
#&gt; LFC &gt; 1.00 (up)    : 502, 2.8%
#&gt; LFC &lt; -1.00 (down) : 247, 1.4%
#&gt; outliers [1]       : 0, 0%
#&gt; low counts [2]     : 0, 0%
#&gt; (mean count &lt; 3)
#&gt; [1] see &#39;cooksCutoff&#39; argument of ?results
#&gt; [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="3-3-model-assays.html#cb34-1"></a>DESeq2<span class="op">::</span><span class="kw">plotMA</span>(res, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>,<span class="dv">10</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:ma-plot"></span>
<img src="img/ma-plot-1.png" alt="Visualization of DESeq2 results as an “MA plot”. Genes that have an adjusted p-value below 0.01 are colored blue.In this case the LFC between conditions is shown on the y-axis, while the average normalised gene counts across all conditions are shown on the x-axis. The assumption of an RNA-seq analysis that most genes are not DE, so most genes are scattered about zero on the y-axis while genes that have evidence of DE are far from the zero baseline. The x-axis gives a sense of the total expression of the gene." width="100%" />
<p class="caption">
Figure 3.3: Visualization of <code>DESeq2</code> results as an “MA plot”. Genes that have an adjusted <em>p-value</em> below 0.01 are colored blue.In this case the LFC between conditions is shown on the y-axis, while the average normalised gene counts across all conditions are shown on the x-axis. The assumption of an RNA-seq analysis that most genes are not DE, so most genes are scattered about zero on the y-axis while genes that have evidence of DE are far from the zero baseline. The x-axis gives a sense of the total expression of the gene.
</p>
</div>

<p>We now output the results as a <em>GRanges</em> object, and due to the conventions of
<strong>plyranges</strong>, we construct a new column called <em>gene_id</em> from the row names of
the results. Each row now contains the genomic region ( <em>seqnames</em>, <em>start</em>,
<em>end</em>, <em>strand</em>) along with corresponding metadata columns (the <em>gene_id</em> and
the results of the test). Note that <strong>tximeta</strong> has correctly identified the
reference genome as “hg38”, and this has also been added to the <strong>GRanges</strong> along
the results columns. This kind of book-keeping is vital once overlap operations
are performed to ensure that <strong>plyranges</strong> is not comparing across incompatible
genomes.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="3-3-model-assays.html#cb35-1"></a><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(plyranges))</span>
<span id="cb35-2"><a href="3-3-model-assays.html#cb35-2"></a>de_genes &lt;-<span class="st"> </span><span class="kw">results</span>(dds,</span>
<span id="cb35-3"><a href="3-3-model-assays.html#cb35-3"></a>                    <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;IFNg&quot;</span>,<span class="st">&quot;naive&quot;</span>),</span>
<span id="cb35-4"><a href="3-3-model-assays.html#cb35-4"></a>                    <span class="dt">lfcThreshold=</span><span class="dv">1</span>,</span>
<span id="cb35-5"><a href="3-3-model-assays.html#cb35-5"></a>                    <span class="dt">format=</span><span class="st">&quot;GRanges&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb35-6"><a href="3-3-model-assays.html#cb35-6"></a><span class="st">  </span><span class="kw">names_to_column</span>(<span class="st">&quot;gene_id&quot;</span>)</span>
<span id="cb35-7"><a href="3-3-model-assays.html#cb35-7"></a>de_genes</span></code></pre></div>
<pre><code>#&gt; GRanges object with 17806 ranges and 7 metadata columns:
#&gt;           seqnames              ranges strand |            gene_id   baseMean
#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;  &lt;numeric&gt;
#&gt;       [1]     chrX 100627109-100639991      - | ENSG00000000003.14    171.571
#&gt;       [2]    chr20   50934867-50958555      - | ENSG00000000419.12    967.751
#&gt;       [3]     chr1 169849631-169894267      - | ENSG00000000457.13    682.433
#&gt;       [4]     chr1 169662007-169854080      + | ENSG00000000460.16    262.963
#&gt;       [5]     chr1   27612064-27635277      - | ENSG00000000938.12   2660.102
#&gt;       ...      ...                 ...    ... .                ...        ...
#&gt;   [17802]    chr10   84167228-84172093      - |  ENSG00000285972.1   10.04746
#&gt;   [17803]     chr6   63572012-63583587      + |  ENSG00000285976.1 4586.34617
#&gt;   [17804]    chr16   57177349-57181390      + |  ENSG00000285979.1   14.29653
#&gt;   [17805]     chr8 103398658-103501895      - |  ENSG00000285982.1   27.76296
#&gt;   [17806]    chr10   12563151-12567351      + |  ENSG00000285994.1    6.60409
#&gt;           log2FoldChange     lfcSE      stat    pvalue      padj
#&gt;                &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;       [1]     -0.2822450 0.3005710   0.00000 1.0000000  1.000000
#&gt;       [2]      0.0391223 0.0859708   0.00000 1.0000000  1.000000
#&gt;       [3]      1.2846179 0.1969067   1.44545 0.1483329  1.000000
#&gt;       [4]     -1.4718762 0.2186916  -2.15772 0.0309493  0.409728
#&gt;       [5]      0.6754781 0.2360530   0.00000 1.0000000  1.000000
#&gt;       ...            ...       ...       ...       ...       ...
#&gt;   [17802]      0.5484518  0.444319         0         1         1
#&gt;   [17803]     -0.0339296  0.188005         0         1         1
#&gt;   [17804]      0.3123477  0.522700         0         1         1
#&gt;   [17805]      0.9945187  1.582373         0         1         1
#&gt;   [17806]      0.2539975  0.595751         0         1         1
#&gt;   -------
#&gt;   seqinfo: 25 sequences (1 circular) from hg38 genome</code></pre>
<p>From this, we can restrict the results to those that meet our FDR threshold and
select (and rename) the metadata columns we are interested in:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="3-3-model-assays.html#cb37-1"></a>de_genes &lt;-<span class="st"> </span>de_genes <span class="op">%&gt;%</span></span>
<span id="cb37-2"><a href="3-3-model-assays.html#cb37-2"></a><span class="st">  </span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span></span>
<span id="cb37-3"><a href="3-3-model-assays.html#cb37-3"></a><span class="st">  </span><span class="kw">select</span>(</span>
<span id="cb37-4"><a href="3-3-model-assays.html#cb37-4"></a>    gene_id,</span>
<span id="cb37-5"><a href="3-3-model-assays.html#cb37-5"></a>    <span class="dt">de_log2FC =</span> log2FoldChange,</span>
<span id="cb37-6"><a href="3-3-model-assays.html#cb37-6"></a>    <span class="dt">de_padj =</span> padj</span>
<span id="cb37-7"><a href="3-3-model-assays.html#cb37-7"></a>  )</span></code></pre></div>
<p>We now wish to extract genes for which there is evidence that the LFC is <em>not</em>
large. We perform this test by specifying an LFC threshold and an alternative
hypothesis (<code>altHypothesis</code>) that the LFC is less than the threshold in
absolute value. In this case, the p-values are taken as maximum of the upper and lower Wald tests under the hypothesis absolute value of the estimated LFC is lower than the threshold. To visualize the result of this test, you can run <code>results</code>
without <code>format="GRanges"</code>, and pass this object to <code>plotMA()</code> as before. We
label these genes as <code>other_genes</code> and later as “non-DE genes”, for comparison
with our <code>de_genes</code> set.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="3-3-model-assays.html#cb38-1"></a>other_genes &lt;-<span class="st"> </span><span class="kw">results</span>(dds,</span>
<span id="cb38-2"><a href="3-3-model-assays.html#cb38-2"></a>                       <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;IFNg&quot;</span>,<span class="st">&quot;naive&quot;</span>),</span>
<span id="cb38-3"><a href="3-3-model-assays.html#cb38-3"></a>                       <span class="dt">lfcThreshold=</span><span class="dv">1</span>,</span>
<span id="cb38-4"><a href="3-3-model-assays.html#cb38-4"></a>                       <span class="dt">altHypothesis=</span><span class="st">&quot;lessAbs&quot;</span>,</span>
<span id="cb38-5"><a href="3-3-model-assays.html#cb38-5"></a>                       <span class="dt">format=</span><span class="st">&quot;GRanges&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb38-6"><a href="3-3-model-assays.html#cb38-6"></a><span class="st">  </span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span></span>
<span id="cb38-7"><a href="3-3-model-assays.html#cb38-7"></a><span class="st">  </span><span class="kw">names_to_column</span>(<span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb38-8"><a href="3-3-model-assays.html#cb38-8"></a><span class="st">  </span><span class="kw">select</span>(</span>
<span id="cb38-9"><a href="3-3-model-assays.html#cb38-9"></a>    gene_id,</span>
<span id="cb38-10"><a href="3-3-model-assays.html#cb38-10"></a>    <span class="dt">de_log2FC =</span> log2FoldChange,</span>
<span id="cb38-11"><a href="3-3-model-assays.html#cb38-11"></a>    <span class="dt">de_padj =</span> padj</span>
<span id="cb38-12"><a href="3-3-model-assays.html#cb38-12"></a>  )</span></code></pre></div>
</div>
<div id="atac-seq-peak-differential-abundance-analysis" class="section level3">
<h3><span class="header-section-number">3.3.2</span> ATAC-seq peak differential abundance analysis</h3>
<p>The following section describes the process we have used for generating a
<em>GRanges</em> object of differential peaks from the ATAC-seq data in <span class="citation">Alasoo et al. (<a href="bibliography.html#ref-alasoo" role="doc-biblioref">2018</a>)</span>. The
code chunks for the remainder of this section are optional.</p>
<p>For assessing differential accessibility, we run <strong>limma</strong> <span class="citation">(Smyth <a href="bibliography.html#ref-Smyth2004" role="doc-biblioref">2004</a>)</span>, and
generate the a summary of LFCs and adjusted p-values for the peaks:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="3-3-model-assays.html#cb39-1"></a><span class="kw">library</span>(limma)</span>
<span id="cb39-2"><a href="3-3-model-assays.html#cb39-2"></a>design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>donor <span class="op">+</span><span class="st"> </span>condition, <span class="kw">colData</span>(atac))</span>
<span id="cb39-3"><a href="3-3-model-assays.html#cb39-3"></a>fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(<span class="kw">assay</span>(atac), design)</span>
<span id="cb39-4"><a href="3-3-model-assays.html#cb39-4"></a>fit &lt;-<span class="st"> </span><span class="kw">eBayes</span>(fit)</span>
<span id="cb39-5"><a href="3-3-model-assays.html#cb39-5"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(fit<span class="op">$</span>coefficients) <span class="op">==</span><span class="st"> &quot;conditionIFNg&quot;</span>)</span>
<span id="cb39-6"><a href="3-3-model-assays.html#cb39-6"></a>tt &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit, <span class="dt">coef=</span>idx, <span class="dt">sort.by=</span><span class="st">&quot;none&quot;</span>, <span class="dt">n=</span><span class="kw">nrow</span>(atac))</span></code></pre></div>
<p>We now take the <code>rowRanges()</code> of the <em>SummarizedExperiment</em> and attach the LFCs
and adjusted p-values from <strong>limma</strong>, so that we can consider the overlap with
differential expression. Note that we set the genome build to “hg38” and
restyle the chromosome information to use the “UCSC” style (e.g. “chr1”,
“chr2”, etc.). Again, we know the genome build from the Zenodo entry for the
ATAC-seq data.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="3-3-model-assays.html#cb40-1"></a>atac_peaks &lt;-<span class="st"> </span><span class="kw">rowRanges</span>(atac) <span class="op">%&gt;%</span></span>
<span id="cb40-2"><a href="3-3-model-assays.html#cb40-2"></a><span class="st">  </span><span class="kw">remove_names</span>() <span class="op">%&gt;%</span></span>
<span id="cb40-3"><a href="3-3-model-assays.html#cb40-3"></a><span class="st">  </span><span class="kw">mutate</span>(</span>
<span id="cb40-4"><a href="3-3-model-assays.html#cb40-4"></a>    <span class="dt">da_log2FC =</span> tt<span class="op">$</span>logFC,</span>
<span id="cb40-5"><a href="3-3-model-assays.html#cb40-5"></a>    <span class="dt">da_padj =</span> tt<span class="op">$</span>adj.P.Val</span>
<span id="cb40-6"><a href="3-3-model-assays.html#cb40-6"></a>  ) <span class="op">%&gt;%</span></span>
<span id="cb40-7"><a href="3-3-model-assays.html#cb40-7"></a><span class="st">  </span><span class="kw">set_genome_info</span>(<span class="dt">genome =</span> <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb40-8"><a href="3-3-model-assays.html#cb40-8"></a></span>
<span id="cb40-9"><a href="3-3-model-assays.html#cb40-9"></a><span class="kw">seqlevelsStyle</span>(atac_peaks) &lt;-<span class="st"> &quot;UCSC&quot;</span></span></code></pre></div>
<p>The final <em>GRanges</em> object containing the DA peaks is included in
the <strong>fluentGenomics</strong> and can be loaded as follows:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="3-3-model-assays.html#cb41-1"></a><span class="kw">library</span>(fluentGenomics)</span>
<span id="cb41-2"><a href="3-3-model-assays.html#cb41-2"></a>peaks</span></code></pre></div>
<pre><code>#&gt; GRanges object with 296220 ranges and 3 metadata columns:
#&gt;            seqnames              ranges strand |          peak_id da_log2FC
#&gt;               &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;character&gt; &lt;numeric&gt;
#&gt;        [1]     chr1          9979-10668      * |      ATAC_peak_1  0.266185
#&gt;        [2]     chr1         10939-11473      * |      ATAC_peak_2  0.322177
#&gt;        [3]     chr1         15505-15729      * |      ATAC_peak_3 -0.574160
#&gt;        [4]     chr1         21148-21481      * |      ATAC_peak_4 -1.147066
#&gt;        [5]     chr1         21864-22067      * |      ATAC_peak_5 -0.896143
#&gt;        ...      ...                 ...    ... .              ...       ...
#&gt;   [296216]     chrX 155896572-155896835      * | ATAC_peak_296216 -0.834629
#&gt;   [296217]     chrX 155958507-155958646      * | ATAC_peak_296217 -0.147537
#&gt;   [296218]     chrX 156016760-156016975      * | ATAC_peak_296218 -0.609732
#&gt;   [296219]     chrX 156028551-156029422      * | ATAC_peak_296219 -0.347678
#&gt;   [296220]     chrX 156030135-156030785      * | ATAC_peak_296220  0.492442
#&gt;                da_padj
#&gt;              &lt;numeric&gt;
#&gt;        [1] 9.10673e-05
#&gt;        [2] 2.03435e-05
#&gt;        [3] 3.41708e-08
#&gt;        [4] 8.22299e-26
#&gt;        [5] 4.79453e-11
#&gt;        ...         ...
#&gt;   [296216] 1.33546e-11
#&gt;   [296217] 3.13015e-01
#&gt;   [296218] 3.62339e-09
#&gt;   [296219] 6.94823e-06
#&gt;   [296220] 7.07664e-13
#&gt;   -------
#&gt;   seqinfo: 23 sequences from hg38 genome; no seqlengths</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="3-2-se.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="3-4-integrate-ranges.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//03-fluentGenomics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
