<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Fluent genomics with plyranges and tximeta | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="Chapter 3 Fluent genomics with plyranges and tximeta | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Fluent genomics with plyranges and tximeta | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Fluent genomics with plyranges and tximeta | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="acknowledgements-1.html"/>
<link rel="next" href="3-1-introduction.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-a-grammar-for-genomic-data-analysis.html"><a href="1-1-a-grammar-for-genomic-data-analysis.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-integration-and-representation-of-genomic-data-structures.html"><a href="1-2-integration-and-representation-of-genomic-data-structures.html"><i class="fa fa-check"></i><b>1.2</b> Integration and representation of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-visual-analytics-for-dimension-reduction-workflows.html"><a href="1-3-visual-analytics-for-dimension-reduction-workflows.html"><i class="fa fa-check"></i><b>1.3</b> Visual analytics for dimension reduction workflows</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <code>plyranges</code></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <code>tximeta</code> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <code>plyranges</code></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Tidy coverage analysis and visualisation with superintronic</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#motivation-intron-signal-in-rna-seq-data"><i class="fa fa-check"></i><b>4.2.1</b> Motivation: intron signal in RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-results-1.html"><a href="4-3-results-1.html"><i class="fa fa-check"></i><b>4.3</b> Results</a><ul>
<li class="chapter" data-level="4.3.1" data-path="4-3-results-1.html"><a href="4-3-results-1.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.3.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.3.2" data-path="4-3-results-1.html"><a href="4-3-results-1.html#integration-of-external-gene-annotations"><i class="fa fa-check"></i><b>4.3.2</b> Integration of external gene annotations</a></li>
<li class="chapter" data-level="4.3.3" data-path="4-3-results-1.html"><a href="4-3-results-1.html#discovery-of-regions-of-interest-via-ranglers"><i class="fa fa-check"></i><b>4.3.3</b> Discovery of regions of interest via ‘ranglers’</a></li>
<li class="chapter" data-level="4.3.4" data-path="4-3-results-1.html"><a href="4-3-results-1.html#a-superintronic-workflow-for-uncovering-intron-retention"><i class="fa fa-check"></i><b>4.3.4</b> A superintronic workflow for uncovering intron retention</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="4.5" data-path="4-5-conclusion-1.html"><a href="4-5-conclusion-1.html"><i class="fa fa-check"></i><b>4.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-design.html"><a href="5-2-design.html"><i class="fa fa-check"></i><b>5.2</b> Design</a><ul>
<li class="chapter" data-level="5.2.1" data-path="5-2-design.html"><a href="5-2-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.2.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.2.2" data-path="5-2-design.html"><a href="5-2-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.2.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.2.3" data-path="5-2-design.html"><a href="5-2-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.2.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="5-3-implementation.html"><a href="5-3-implementation.html"><i class="fa fa-check"></i><b>5.3</b> Implementation</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-implementation.html"><a href="5-3-implementation.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.3.1</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-implementation.html"><a href="5-3-implementation.html#linking-views-via-brushes"><i class="fa fa-check"></i><b>5.3.2</b> Linking views via brushes</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-case-studies.html"><a href="5-4-case-studies.html"><i class="fa fa-check"></i><b>5.4</b> Case Studies</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-case-studies.html"><a href="5-4-case-studies.html#case-study-1-exploring-tree-structured-data"><i class="fa fa-check"></i><b>5.4.1</b> Case Study 1: Exploring tree structured data</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-case-studies.html"><a href="5-4-case-studies.html#case-study-2-clustering-pbmc-10x-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.4.2</b> Case Study 2: Clustering PBMC 10x single cell RNA-seq data</a></li>
<li class="chapter" data-level="5.4.3" data-path="5-4-case-studies.html"><a href="5-4-case-studies.html#case-study-3-single-cell-mouse-retina-data"><i class="fa fa-check"></i><b>5.4.3</b> Case Study 3: Single cell mouse retina data</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-discussion-3.html"><a href="5-5-discussion-3.html"><i class="fa fa-check"></i><b>5.5</b> Discussion</a></li>
<li class="chapter" data-level="5.6" data-path="5-6-acknowledgements-3.html"><a href="5-6-acknowledgements-3.html"><i class="fa fa-check"></i><b>5.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-findings-and-limitations.html"><a href="6-1-findings-and-limitations.html"><i class="fa fa-check"></i><b>6.1</b> Findings and limitations</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-software-development-and-impact.html"><a href="6-2-software-development-and-impact.html"><i class="fa fa-check"></i><b>6.2</b> Software development and impact</a></li>
<li class="chapter" data-level="6.3" data-path="6-3-further-research.html"><a href="6-3-further-research.html"><i class="fa fa-check"></i><b>6.3</b> Further research</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:fluentGenomics" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></h1>
<p>We construct a simple workflow for fluent genomics data analysis using the R/Bioconductor ecosystem. This involves three core steps: import the data into an appropriate abstraction, model the data with respect to the biological questions of interest, and integrate the results with respect to their underlying genomic coordinates. Here we show how to implement these steps to integrate published RNA-seq and ATAC-seq experiments on macrophage cell lines. Using tximeta, we import RNA-seq transcript quantifications into an analysis-ready data structure, called the SummarizedExperiment, that contains the ranges of the reference transcripts and metadata on their provenance. Using SummarizedExperiments to represent the ATAC-seq and RNA-seq data, we model differentially accessible (DA) chromatin peaks and differentially expressed (DE) genes with existing Bioconductor packages. Using plyranges we then integrate the results to see if there is an enrichment of DA peaks near DE genes by finding overlaps and aggregating over log-fold change thresholds. The combination of these packages and their integration with the Bioconductor ecosystem provide a coherent framework for analysts to iteratively and reproducibly explore their biological data.</p> 
</div>
            </section>

          </div>
        </div>
      </div>
<a href="acknowledgements-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="3-1-introduction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//03-fluentGenomics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
