<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.3 A workflow for uncovering intron retention in a zebrafish experiment | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="4.3 A workflow for uncovering intron retention in a zebrafish experiment | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="4.3 A workflow for uncovering intron retention in a zebrafish experiment | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.3 A workflow for uncovering intron retention in a zebrafish experiment | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="4-2-methods.html"/>
<link rel="next" href="4-4-discussion-2.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="0-1-open-research.html"><a href="0-1-open-research.html"><i class="fa fa-check"></i><b>0.1</b> Open Research</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-sec-tidygranges.html"><a href="1-1-sec-tidygranges.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-sec-integrate.html"><a href="1-2-sec-integrate.html"><i class="fa fa-check"></i><b>1.2</b> Integration of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-sec-coverage.html"><a href="1-3-sec-coverage.html"><i class="fa fa-check"></i><b>1.3</b> Representation of genomic data structures</a></li>
<li class="chapter" data-level="1.4" data-path="1-4-sec-va.html"><a href="1-4-sec-va.html"><i class="fa fa-check"></i><b>1.4</b> Interactive visualisation for high-dimensional data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <code>plyranges</code></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <code>tximeta</code> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <code>plyranges</code></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Exploratory coverage analysis with superintronic and plyranges</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.2.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.2.2" data-path="4-2-methods.html"><a href="4-2-methods.html#integration-of-external-annotations"><i class="fa fa-check"></i><b>4.2.2</b> Integration of external annotations</a></li>
<li class="chapter" data-level="4.2.3" data-path="4-2-methods.html"><a href="4-2-methods.html#discovery-of-regions-of-interest-via-data-descriptors"><i class="fa fa-check"></i><b>4.2.3</b> Discovery of regions of interest via ‘data descriptors’</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><i class="fa fa-check"></i><b>4.3</b> A workflow for uncovering intron retention in a zebrafish experiment</a></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="4.5" data-path="4-5-acknowledgements-3.html"><a href="4-5-acknowledgements-3.html"><i class="fa fa-check"></i><b>4.5</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-dimensionality-reduction-background.html"><a href="5-2-dimensionality-reduction-background.html"><i class="fa fa-check"></i><b>5.2</b> Dimensionality Reduction Background</a></li>
<li class="chapter" data-level="5.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html"><i class="fa fa-check"></i><b>5.3</b> Visual Design</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.3.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.3.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.3.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.3.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html"><i class="fa fa-check"></i><b>5.4</b> Software Infrastructure</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-explore-the-subspace-of-d-dimensional-projections"><i class="fa fa-check"></i><b>5.4.1</b> Tours explore the subspace of <span class="math inline">\(d\)</span>-dimensional projections</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.4.2</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.4.3" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#linking-and-highlighting-views-via-interactions"><i class="fa fa-check"></i><b>5.4.3</b> Linking and highlighting views via interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html"><i class="fa fa-check"></i><b>5.5</b> Case Studies</a><ul>
<li class="chapter" data-level="5.5.1" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-1-exploring-simulation-data-with-tours-and-t-sne"><i class="fa fa-check"></i><b>5.5.1</b> Case Study 1: Exploring simulation data with tours and t-SNE</a></li>
<li class="chapter" data-level="5.5.2" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-2-clustering-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.5.2</b> Case Study 2: Clustering single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="5-6-discussion-3.html"><a href="5-6-discussion-3.html"><i class="fa fa-check"></i><b>5.6</b> Discussion</a></li>
<li class="chapter" data-level="5.7" data-path="5-7-acknowledgements-4.html"><a href="5-7-acknowledgements-4.html"><i class="fa fa-check"></i><b>5.7</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-software-development.html"><a href="6-1-software-development.html"><i class="fa fa-check"></i><b>6.1</b> Software Development</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-further-work.html"><a href="6-2-further-work.html"><i class="fa fa-check"></i><b>6.2</b> Further Work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment" class="section level2">
<h2><span class="header-section-number">4.3</span> A workflow for uncovering intron retention in a zebrafish experiment</h2>
<p>In the study of gene regulation, there is much interest in uncovering the
effects of aberrant minor class splicing (called U12 splicing) on the transcriptome.
Minor class splicing is a regulatory process where a class of
introns (called U12 introns in this case but there exist other classes in eukaryotic organisms such as U2 introns) are removed from pre-mRNA prior to gene expression. The removal of these introns is catalysed by small ribonucleoproteins (snRNPs)
which identify key motifs and branch sites in the intron to begin splicing <span class="citation">(Markmiller et al. <a href="bibliography.html#ref-Markmiller2014-kg">2014</a>)</span>.
Here we describe an exploratory workflow for finding coverage traces
with evidence for intron retention using RNA-seq data from a knockout experiment in zebrafish obtained from the Heath Lab at WEHI.</p>
<p>The data consist of 11 zebrafish samples from single-end polyA enriched RNA-seq libraries pooled from zebrafish larvae. The experimental factors looked at combinations of genotype (whether the gene rnpc3 has been knocked out or not) and line (whether the zebrafish larvae have the caliban <em>cal</em> or mutant caliban <em>zm-cal</em> phenotype). Within each combination there are three biological replicates, except for the combination <em>zm-cal</em> and wild-type rnpc3 which
had two replicates.</p>
<p>FASTQ files were aligned to the GRCz11 reference genome using subjunc with
the default parameters called from Rsubread to produce BAM files for each
sample <span class="citation">(Liao, Smyth, and Shi <a href="bibliography.html#ref-Liao2013-eb">2013</a>, <a href="bibliography.html#ref-Liao2019-qn">2019</a>)</span>. The coverage was then estimated directly from the set of BAM files
using <strong>superintronic</strong> into the long form <em>GRanges</em> representation we described above.</p>
<p>The gene annotation files were
obtained as GFF files from RefSeq and used that to construct the exonic
and intronic parts of each gene as <strong>GRanges</strong> object using <em>superintronic</em>.
We further filtered genes that had a single exon or genes that were overlapping
others in the annotation and that were not on the main contigs of the reference genome (i.e. excluding mitochondria) to simplify our analysis and reduce any coverage ambiguity. This left 18,270 genes available for
computing data descriptors on in order to detecting IR-like coverage traces. Across each combination of genotype and line, we first log<span class="math inline">\(_2\)</span> transformed the coverage score with an offset of one, and then intersected the coverage <em>GRanges</em> with exonic and intronic features of each gene. For each gene,
the weighted mean and standard deviation of the log-transformed coverage score were computed over all intron and exon parts within the groupings of
genotype and line. We also computed the bases above descriptor for intron
features. It refers to the total number of bases within an intron that has a score above the overall average exon coverage score. The scatter plot
matrix view of these descriptors is shown in figure <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-splom">4.2</a>
for a single biological group in the experiment. Using these views and
by summarising over the data descriptors, we came up with thresholds
for finding genes that have IR-like traces within each biological group.</p>

<div class="figure" style="text-align: center"><span id="fig:si-splom"></span>
<img src="img/Cal-Rnpc3_KO-pairs-plot.png" alt="A hexbin scatter plot matrix of the data descriptors estimated for the cal rnpc3 knock out zebrafish line. To identify coverage traces with IR like events, we want a set of descriptors that will find genes with the following characteristics: the gene is “expressed” that has a large number of intron bases relative to the coverage of other intron features, and has relatively stable coverage within exon features. To do this we looked at the descriptor, as well as computing the the mean and standard deviation of both exon and intron features." width="80%" />
<p class="caption">
Figure 4.2: A hexbin scatter plot matrix of the data descriptors estimated for the <em>cal</em> rnpc3 knock out zebrafish line. To identify coverage traces with IR like events, we want a set of descriptors that will find genes with the following characteristics: the gene is “expressed” that has a large number of intron bases relative to the coverage of other intron features, and has relatively stable coverage within exon features. To do this we looked at the descriptor, as well as computing the the mean and standard deviation of both exon and intron features.
</p>
</div>
<p>We decided to select genes with the following thresholds: genes have an average exon log-coverage greater than the mean of average exon log-coverage values across all genes, have an average intron log-coverage greater than the average standard deviation of exon log-coverage values over all genes, and
the standard deviation of intron log-coverage values is twice the standard
deviation of exon log-coverage values. That is, we are selecting genes that
are expressed but have large average intron expression that is more variable than the gene’s exon expression. This results in a total of 86 genes selected
to link back to their underlying coverage traces, with the overlaps shown
in the UpSet plot in figure <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-upset">4.3</a> <span class="citation">(Lex et al. <a href="bibliography.html#ref-Lex2014-bh">2014</a>)</span>. The procedure produces gene coverage traces with known minor class splice sites affected by the knockout procedure such as ccdc43 and nat15 (figures <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-cvg-ccdc43">4.4</a> and <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-cvg-nat15">4.5</a>), as well as some that appear to have U12 intron retention like events such as mapk3 or tspan31 (figures <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-cvg-mapk3">4.6</a> and <a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html#fig:si-cvg-tspan31">4.7</a>) that affect other parts of the gene.</p>

<div class="figure" style="text-align: center"><span id="fig:si-upset"></span>
<img src="img/si-olaps.png" alt="Gene overlaps found between each combination of genotype and line using the thresholds defined in the text. Our procedure mostly finds genes with IR like profiles across all groups (28 shared between all four) or that is unique to a single group, since we do not consider looking at differential IR and run our thresholding separately for each group of replicates. The rnpc3 knockout lines" width="80%" />
<p class="caption">
Figure 4.3: Gene overlaps found between each combination of genotype and line using the thresholds defined in the text. Our procedure mostly finds genes with IR like profiles across all groups (28 shared between all four) or that is unique to a single group, since we do not consider looking at differential IR and run our thresholding separately for each group of replicates. The rnpc3 knockout lines
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:si-cvg-ccdc43"></span>
<img src="img/ccdc43.pdf" alt="Coverage traces faceted by biological group for the zebrafish knockout experiment, oriented from the 5’ to 3’ end of the gene. Orange areas refer to intronic parts of the gene, while dark green areas refer to the exonic part. The ccdc43 gene is known to have enhanced U12 intron retention in the caliban phenotype and increase retention when rnpc3 is knocked out as can be seen directly from the intron located at around 31.603Mb." width="80%" />
<p class="caption">
Figure 4.4: Coverage traces faceted by biological group for the zebrafish knockout experiment, oriented from the 5’ to 3’ end of the gene. Orange areas refer to intronic parts of the gene, while dark green areas refer to the exonic part. The ccdc43 gene is known to have enhanced U12 intron retention in the caliban phenotype and increase retention when rnpc3 is knocked out as can be seen directly from the intron located at around 31.603Mb.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:si-cvg-nat15"></span>
<img src="img/nat15.pdf" alt="The gene nat15 exhibits another example of U12 intron retention (located at around 27.93Mb) in the rnpc3 knockout groups." width="80%" />
<p class="caption">
Figure 4.5: The gene nat15 exhibits another example of U12 intron retention (located at around 27.93Mb) in the rnpc3 knockout groups.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:si-cvg-mapk3"></span>
<img src="img/mapk3.pdf" alt="The gene mapk3 appears to have intron retention close to the start of the gene that have different impacts downstream, which result in different IR profiles between the knockout and wild-type groups." width="80%" />
<p class="caption">
Figure 4.6: The gene mapk3 appears to have intron retention close to the start of the gene that have different impacts downstream, which result in different IR profiles between the knockout and wild-type groups.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:si-cvg-tspan31"></span>
<img src="img/tspan31.pdf" alt="The gene tspan31 has a potential cryptic splice site within the intron retained in the knockout groups." width="80%" />
<p class="caption">
Figure 4.7: The gene tspan31 has a potential cryptic splice site within the intron retained in the knockout groups.
</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="4-2-methods.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="4-4-discussion-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//04-intron.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
