<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Tidy coverage analysis uncovers intron retention | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="Chapter 4 Tidy coverage analysis uncovers intron retention | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Tidy coverage analysis uncovers intron retention | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Tidy coverage analysis uncovers intron retention | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="3-7-acknowledgements-2.html"/>
<link rel="next" href="5-ch-tsne.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-acknowledgements-1.html"><a href="2-5-acknowledgements-1.html"><i class="fa fa-check"></i><b>2.5</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.6" data-path="2-6-availability-of-data-and-materials.html"><a href="2-6-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.6</b> Availability of Data and Materials</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="3.7" data-path="3-7-acknowledgements-2.html"><a href="3-7-acknowledgements-2.html"><i class="fa fa-check"></i><b>3.7</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Tidy coverage analysis uncovers intron retention</a></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Spurious embeddings from non-linear dimensionality reduction methods and how to find them</a></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion and future plans</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:intron" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Tidy coverage analysis uncovers intron retention</h1>
<!-- # Introduction -->
<!-- ## Data -->
<!-- We will use data from 12 samples from the human cell line mixture RNA-seq -->
<!-- data. There are three biological replicates across two library preparation kits, -->
<!-- a polyA kit and a total RNA kit. The BAM files have been previously aligned -->
<!-- with subread using the hg20 reference genome.  -->
<!-- ```{r prepare-design, echo= FALSE} -->
<!-- design <- readRDS(here("data", "superintronic", "design.rds")) -->
<!-- ``` -->
<!-- ## `superintronic` analysis -->
<!-- A superintronic analysis consists of four steps: -->
<!-- 1. Preparing the annotation and extract exonic/intronic parts -->
<!-- 2. Computing coverage over regions of interest -->
<!-- 3. Summarising coverage over regions of interest -->
<!-- 4. Visualising the results -->
<!-- Steps 1 and 2. are performed in the `scripts/run_coverage.R` -->
<!-- ### Preparing GFF/GTF files -->
<!-- ```{r gff} -->
<!-- library(superintronic) -->
<!-- parts <- readRDS(here::here("data", "complete-annotation.rds")) -->
<!-- parts -->
<!-- ``` -->
<!-- These can be interrogated as desired, but for our purposes we will keep -->
<!-- genes that are protein coding, on main chromosomes but not mitochondrial  -->
<!-- genome, and do not overlap any other genes and have more than one exon. -->
<!-- ```{r default-filter, echo = FALSE} -->
<!-- parts_sub <- readRDS(here::here("data", "filtered-annotation.rds")) -->
<!-- parts_sub -->
<!-- ``` -->
<!-- ## Coverage over intronic/exonic regions -->
<!-- Now all the ingredients are in place to merge the coverage scores -->
<!-- to our prepared annotation with `join_parts()`. This again -->
<!-- returns a GRanges object, restricted to the intersection of the  -->
<!-- coverage ranges with intron/exon ranges. Additional columns are added -->
<!-- corresponding to the properties of the intron/exon. -->
<!-- ```{r, load-features, echo = FALSE } -->
<!-- cvg_over_features <- readRDS(here::here("data", "parts-coverage.rds")) -->
<!-- ``` -->
<!-- Now we will compute summaries for each cell line and kit to detect IR  -->
<!-- genes. This consists of three steps -->
<!-- 1. Computing the average log-coverage over all exon/intron features: -->
<!-- ```{r rangenostics-01} -->
<!-- # compute intron/exon features -->
<!-- cvg_over_features <- cvg_over_features %>%  -->
<!--   mutate(log_score = log2(score + 0.5))   -->
<!-- cvg_by_features <- cvg_over_features %>%  -->
<!--   group_by(Kit, CellLine, gene_id, feature_type) -->
<!-- E_vals <- cvg_by_features %>%  -->
<!--   summarise(mn = Hmisc::wtd.mean(log_score, width), -->
<!--             sd = sqrt(Hmisc::wtd.var(log_score, width)), -->
<!--             raw_mn = Hmisc::wtd.mean(score, width), -->
<!--             score = score, -->
<!--             n_bases = width, -->
<!--             start = min(start), -->
<!--             end = max(end), -->
<!--             seqnames = unlist(unique(seqnames)), -->
<!--             strand = unlist(unique(feature_strand))) %>%  -->
<!--   makeGRangesFromDataFrame(keep.extra.columns = TRUE) -->
<!-- ``` -->
<!-- 2. For exon features computing the mean and variance of average log-coverage  -->
<!--    over all genes (lumpiness). -->
<!-- ```{r rangenostics-02} -->
<!-- lumpy <- E_vals %>%  -->
<!--   filter(sd > 0, feature_type == "exon") %>%  -->
<!--   group_by(Kit, CellLine) %>%  -->
<!--   summarise( -->
<!--     E_mu = mean(mn), -->
<!--     E_sd = mean(sd), -->
<!--     E_raw = mean(raw_mn) -->
<!-- ) -->
<!-- lumpy  -->
<!-- mcols(E_vals) <- cbind(mcols(E_vals), -->
<!--                        lumpy[match(paste0(E_vals$Kit, E_vals$CellLine),  -->
<!--                                    paste0(lumpy$Kit, lumpy$CellLine)), -c(1,2)]) -->
<!-- E_vals <- E_vals %>%  -->
<!--   mutate( -->
<!--     n_bases_E_raw = sum(n_bases[score > E_raw]), -->
<!--   )  -->
<!-- ``` -->
<!-- 3. Compute the average number of bases with a  -->
<!--    log-coverage score above the E values defined above. We have also computed -->
<!--    a smooth between over the intron/exon feature standard deviation against -->
<!--    the average for visualisation purposes. -->
<!-- ```{r rangenostics-03} -->
<!-- rango <- E_vals %>%  -->
<!--   plyranges::select(-n_bases, -score, -seqnames, -start, -end, -width, -strand,  -->
<!--          .drop_ranges = TRUE) %>%  -->
<!--   dplyr::as_tibble() %>%  -->
<!--   group_by(Kit, CellLine, feature_type) %>%  -->
<!--   tidyr::nest() %>%  -->
<!--   mutate( -->
<!--     smooth = lapply(data, function(x) { -->
<!--       mgcv::gam(sd ~ s(mn), data = x) -->
<!--       }), -->
<!--     augment = lapply(smooth, broom::augment), -->
<!--   ) %>%  -->
<!--   tidyr::unnest(data, augment) %>%  -->
<!--   dplyr::select(-dplyr::ends_with("1"), -.sigma) -->
<!-- ``` -->
<!-- 4. Plot the results. -->
<!-- We get very similar results to a `voom` plot in a standard limma analysis. We -->
<!-- have also overlaid on the E_mu and E_sigma values as potential cut-offs for -->
<!-- each feature.  -->
<!-- ```{r voom-like-plot} -->
<!-- # rangenostics feature_average vs feature_sd coloured by n of bases above -->
<!-- # average raw exon coverage -->
<!-- library(ggplot2) -->
<!-- voom_plot <- function(.x, .y) { -->
<!--   p <- ggplot(data = .x, aes(x = mn, y = sd)) + -->
<!--     geom_point() + -->
<!--     geom_line(aes(y = .fitted), colour = "blue") + -->
<!--     geom_vline(aes(xintercept = E_mu),  -->
<!--                data  = dplyr::distinct(.x, feature_type, E_mu)) + -->
<!--     geom_hline(aes(yintercept = E_sd), -->
<!--                data = dplyr::distinct(.x, feature_type, E_sd)) + -->
<!--     facet_wrap(~feature_type) + -->
<!--     labs( -->
<!--          subtitle = paste(.y$Kit, .y$CellLine), -->
<!--          x = "mean log-coverage", -->
<!--          y = "sd log-covreage" -->
<!--     ) -->
<!--   fname <- here::here("img",  -->
<!--                       paste0(.y$Kit, "-", .y$CellLine,  -->
<!--                              "voom-like-plot.png") -->
<!--   ) -->
<!--   ggsave(fname, p) -->
<!-- } -->
<!-- rango %>%  -->
<!--   group_by(Kit, CellLine) %>%  -->
<!--   dplyr::group_walk(voom_plot) -->
<!-- ``` -->
<!-- We also plot pairs of summary values for each feature in a scatter plot matrix -->
<!-- (using a hexbin scatter plot). Again we get similar results to the plots -->
<!-- produced by a DE analysis from InDeX. -->
<!-- ```{r rangenostics-pairs} -->
<!-- all_features <- tidyr::gather(rango, "key", "value",  -->
<!--                               -Kit, -CellLine, -gene_id, -feature_type,  -->
<!--                               -E_mu, -E_sd, -E_raw) %>%  -->
<!--   mutate(var = paste0(feature_type, "_", key)) %>%  -->
<!--   plyranges::select(-feature_type, -key) %>%  -->
<!--   arrange(gene_id) %>%  -->
<!--   tidyr::spread(var, value) -->
<!-- pairs_plot <- function(.x, .y) { -->
<!--   sub <- plyranges::select(.x,  -->
<!--                            exon_mn,  -->
<!--                            intron_mn,  -->
<!--                            intron_sd,  -->
<!--                            bases_above = intron_n_bases_E_raw) -->
<!--   fname <- here::here("img",  -->
<!--                       paste0(.y$Kit, "-", .y$CellLine,  -->
<!--                              "-pairs-plot.png") -->
<!--   ) -->
<!--   p <- GGally::ggpairs(sub, columns = 1:4, lower = list(continuous = hex)) -->
<!--   ggsave(fname, p) -->
<!-- } -->
<!-- all_features %>%  -->
<!--   group_by(Kit, CellLine) %>%  -->
<!--   dplyr::group_walk(pairs_plot) -->
<!-- ``` -->
<!-- 5. Decide on cut-off over each feature. It appears that choosing genes -->
<!-- with high intron feature standard deviations, and with number of bases above -->
<!-- E_mu in the 99th percentile discriminates intron retention features well. -->
<!-- ```{r rangenositcs-hits} -->
<!-- hits <- all_features %>%  -->
<!--   group_by(Kit, CellLine) %>%  -->
<!--   filter(exon_mn > E_mu,  -->
<!--          intron_sd > E_sd,  -->
<!--          intron_n_bases_E_raw > quantile(intron_n_bases_E_raw, 0.99))   -->
<!-- dplyr::count(hits) -->
<!-- ``` -->
<!-- 6. We can also look at overlaps between sets using an UpSet plot: -->
<!-- ```{r, echo  = FALSE} -->
<!-- sets <- hits %>%  -->
<!--   dplyr::distinct(gene_id) %>% -->
<!--   arrange(gene_id) %>%  -->
<!--   mutate(value = 1L, var = paste(CellLine, Kit)) %>%  -->
<!--   ungroup() %>%   -->
<!--   plyranges::select(gene_id, var, value) %>%  -->
<!--   tidyr::spread(var, value) %>%  -->
<!--   dplyr::mutate_at(2:5, .funs = ~ dplyr::if_else(is.na(.), 0L, .))  -->
<!-- sets <- as.data.frame(sets) -->
<!-- library(UpSetR) -->
<!-- pdf(here::here("img", "olaps-superintronic-hits.pdf")) -->
<!-- upset(sets) -->
<!-- dev.off() -->
<!-- ``` -->

</div>
            </section>

          </div>
        </div>
      </div>
<a href="3-7-acknowledgements-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="5-ch-tsne.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//04-intron.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
