<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.3 Visual Design | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="5.3 Visual Design | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="5.3 Visual Design | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.3 Visual Design | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="5-2-dimensionality-reduction.html"/>
<link rel="next" href="5-4-software-infrastructure.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-sec-tidygranges.html"><a href="1-1-sec-tidygranges.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-sec-ir.html"><a href="1-2-sec-ir.html"><i class="fa fa-check"></i><b>1.2</b> Integration and representation of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-sec-va.html"><a href="1-3-sec-va.html"><i class="fa fa-check"></i><b>1.3</b> Visual analytics for dimension reduction workflows</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <code>plyranges</code></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <code>tximeta</code> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <code>plyranges</code></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Exploratory coverage analysis with superintronic and plyranges</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.2.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.2.2" data-path="4-2-methods.html"><a href="4-2-methods.html#integration-of-external-annotations"><i class="fa fa-check"></i><b>4.2.2</b> Integration of external annotations</a></li>
<li class="chapter" data-level="4.2.3" data-path="4-2-methods.html"><a href="4-2-methods.html#discovery-of-regions-of-interest-via-data-descriptors"><i class="fa fa-check"></i><b>4.2.3</b> Discovery of regions of interest via ‘data descriptors’</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><i class="fa fa-check"></i><b>4.3</b> A workflow for uncovering intron retention in a zebrafish experiment</a></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="4.5" data-path="4-5-acknowledgements-3.html"><a href="4-5-acknowledgements-3.html"><i class="fa fa-check"></i><b>4.5</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-dimensionality-reduction.html"><a href="5-2-dimensionality-reduction.html"><i class="fa fa-check"></i><b>5.2</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="5.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html"><i class="fa fa-check"></i><b>5.3</b> Visual Design</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.3.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.3.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.3.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.3.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html"><i class="fa fa-check"></i><b>5.4</b> Software Infrastructure</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.4.1</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#linking-views-via-brushes"><i class="fa fa-check"></i><b>5.4.2</b> Linking views via brushes</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html"><i class="fa fa-check"></i><b>5.5</b> Case Studies</a><ul>
<li class="chapter" data-level="5.5.1" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-1-exploring-tree-structured-data-with-tours-and-t-sne"><i class="fa fa-check"></i><b>5.5.1</b> Case Study 1: Exploring tree structured data with tours and t-SNE</a></li>
<li class="chapter" data-level="5.5.2" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-2-clustering-pbmc-10x-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.5.2</b> Case Study 2: Clustering PBMC 10x single cell RNA-seq data</a></li>
<li class="chapter" data-level="5.5.3" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-3-single-cell-mouse-retina-data"><i class="fa fa-check"></i><b>5.5.3</b> Case Study 3: Single cell mouse retina data</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="5-6-discussion-3.html"><a href="5-6-discussion-3.html"><i class="fa fa-check"></i><b>5.6</b> Discussion</a></li>
<li class="chapter" data-level="5.7" data-path="5-7-acknowledgements-4.html"><a href="5-7-acknowledgements-4.html"><i class="fa fa-check"></i><b>5.7</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-findings-and-impact.html"><a href="6-1-findings-and-impact.html"><i class="fa fa-check"></i><b>6.1</b> Findings and Impact</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-further-research.html"><a href="6-2-further-research.html"><i class="fa fa-check"></i><b>6.2</b> Further Research</a><ul>
<li class="chapter" data-level="6.2.1" data-path="6-2-further-research.html"><a href="6-2-further-research.html#scalable-grammars-for-non-rectangular-data-structures"><i class="fa fa-check"></i><b>6.2.1</b> Scalable grammars for non-rectangular data structures</a></li>
<li class="chapter" data-level="6.2.2" data-path="6-2-further-research.html"><a href="6-2-further-research.html#extending-tour-displays-for-large-n-data-sets"><i class="fa fa-check"></i><b>6.2.2</b> Extending tour displays for large <span class="math inline">\(n\)</span> data sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visual-design" class="section level2">
<h2><span class="header-section-number">5.3</span> Visual Design</h2>
<p>We propose using tours as part of an analyst’s workflow in in performing
dimensionality reduction tasks. We have made extensive use of ensemble
graphics, that is aligning related plots alongside each other to provide
context. As we will see in the case studies, this allows analysts to quickly
compare views from embedding methods and allows them to see how the embedding
method alters the global structure of their data. Using ensembles allows the
use of interaction techniques, that allow analysts to perform cluster
orientation tasks via linking multiple views. This approach allows our
interface, to achieve the three principles for interactive high-dimensional
data visualisation outlined by <span class="citation">Buja, Cook, and Swayne (<a href="bibliography.html#ref-Buja1996-fk">1996</a>)</span>: finding gestalt, posing queries,
and making comparisons.</p>
<div id="finding-gestalt-focus-and-context" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Finding Gestalt: focus and context</h3>
<p>To investigate latent structure and the shape of a high dimensional dataset, a
tour can be run without the use of an external embedding. It is often useful to
first run principal components on the input as an initial dimension reduction
step, and then tour a subset of those components instead, i.e. by selecting
them from a scree plot. The default tour layout is a scatter plot with an axis
layout displaying the magnitude and direction of each basis vector. Since the
tour is dynamic, it is often useful to be able to pause and highlight a
particular view. In our interface, brushing will pause the tour path, allowing
users to identify ‘interesting’ projections. The domain of the axis scales from
running a tour is called the half range, and is computed by rescaling the input
data onto hyper-dimensional unit cube. We bind the half range to a mouse wheel
event, allowing a user to pan and zoom on the tour view dynamically. This is
useful for peeling back dense clumps of points to reveal structure.</p>
</div>
<div id="posing-queries-multiple-views-many-contexts" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Posing Queries: multiple views, many contexts</h3>
<p>We have combined the tour view in a side by side layout with
a scatter plot view that represents the output of an embedding algorithm.
These views are linked; analysts can brush regions or
highlight collections of points in either view. Linked highlighting can
be performed when points have been previously labeled according to some
discrete structure, i.e. cluster labels are available. This is achieved
via the analyst clicking on groups in the legend, which causes unselected groupings to have their points become less opaque. Consequently,
simple linked highlighting can alleviate a known downfall of methods such as
UMAP or t-SNE: that is distances between clusters are misleading. By
highlighting corresponding clusters in the tour view, the analyst can
see the relationship between clusters, and therefore obtain a more accurate
representation of the topology of their data.</p>
<p>Simple linked brushing is achieved via mouse-click and drag movements. By
default, when brushing occurs in the tour view, the current projection is
paused and corresponding points in the embedding view are highlighted.
Likewise, when brushing occurs in the embedding view, corresponding points
in the tour view are highlighted. In this case, an analyst can use brushing
for manually identifying clusters and verifying cluster locations and shapes:
brushing in the embedding view gives analysts a sense of the shape and
proximity of cluster in high-dimensional space.</p>
</div>
<div id="making-comparisons-revising-embeddings" class="section level3">
<h3><span class="header-section-number">5.3.3</span> Making comparisons: revising embeddings</h3>
<p>As mentioned previously, when using any DR method, we are assuming
the embedding is representative of the high-dimensional dataset it was
computed from. Defining what it means for embedding to be ‘representative` or
’faithful’ to high-dimensional data is ill-posed and depends on the underlying
task an analyst is trying to achieve. At the very minimum, we are interested
in distortions and diffusions of the high-dimensional data. Distortions
occur when points that are near each other in the embedding view are
far from each other in the original dataset. This implies that the embedding
is not continuous. Diffusions occur when points are far from each other in the embedding view are near in the original data. Whether, points are near or far
is reliant on the distance metric used; distortions and diffusions can
be thought of as the preservation of distances or the nearest neighbors
graphs between the high-dimensional space and the embedding space. As distances
can be noisy in high-dimensions, ranks can be used instead as has been
proposed by … Identifying distortions and diffusions allows an analyst
to investigate the quality of their embedding and revise them iteratively. We
propose this can be done visually using our side-by-side tour and embedding
views. Some quality checks, such as finding distortions can be identified using simple linked brushing, however we can also interrogate them via spatial
brushes and brush composition.</p>
<p>Look up nearest neighbours graph from points that lie in a brushing region.</p>
<p>Highlight the corresponding neighbours using colour or transparency in the linked view.</p>
<ul>
<li>The <span class="math inline">\(k\)</span> nearest neighbours graph can be pre-computed quickly, for either <span class="math inline">\(X\)</span>, <span class="math inline">\(Y\)</span> or both.</li>
<li>Instead of using the neighbour indices, we could use the neighbour distances instead.</li>
<li>Composition of multiple brushes could be used to show where there are matches/mismatches between nearest neighbour graphs.</li>
</ul>
<p>Using a linked neighbourhood brush, we can
visually investigate the nearest neighbour relationships in the
high-dimensional space via brushing in the embedding view. The user can select
the number of nearest neighbours directly, and modify the distance metric
used for determining the neighbours. This allows users to interrogate the stability of clusters generated in the embedding view. Multiple brushes
can be used to pose queries in either the tour view or the embedding view;
interface controls allow these brushes to combine using logical operators
such as ‘and’, ‘or’, or ‘not’. The use of linked brushing goes beyond
simple color highlights, allowing analysts to get a more holistic
view of the effect of an embedding algorithm.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="5-2-dimensionality-reduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="5-4-software-infrastructure.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//05-embeddings.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
