<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.5 Case Studies | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="5.5 Case Studies | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="5.5 Case Studies | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.5 Case Studies | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="5-4-software-infrastructure.html"/>
<link rel="next" href="5-6-discussion-3.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-sec-tidygranges.html"><a href="1-1-sec-tidygranges.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-sec-integrate.html"><a href="1-2-sec-integrate.html"><i class="fa fa-check"></i><b>1.2</b> Integration of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-sec-coverage.html"><a href="1-3-sec-coverage.html"><i class="fa fa-check"></i><b>1.3</b> Representation of genomic data structures</a></li>
<li class="chapter" data-level="1.4" data-path="1-4-sec-va.html"><a href="1-4-sec-va.html"><i class="fa fa-check"></i><b>1.4</b> Interactive visualisation for high-dimensional data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <strong>plyranges</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <strong>plyranges</strong> and <strong>tximeta</strong></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <strong>tximeta</strong> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <strong>plyranges</strong></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Exploratory coverage analysis with superintronic and plyranges</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.2.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.2.2" data-path="4-2-methods.html"><a href="4-2-methods.html#integration-of-external-annotations"><i class="fa fa-check"></i><b>4.2.2</b> Integration of external annotations</a></li>
<li class="chapter" data-level="4.2.3" data-path="4-2-methods.html"><a href="4-2-methods.html#discovery-of-regions-of-interest-via-data-descriptors"><i class="fa fa-check"></i><b>4.2.3</b> Discovery of regions of interest via ‘data descriptors’</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><i class="fa fa-check"></i><b>4.3</b> A workflow for uncovering intron retention in a zebrafish experiment</a></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-3.html"><a href="acknowledgements-3.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-overview-of-dimension-reduction.html"><a href="5-2-overview-of-dimension-reduction.html"><i class="fa fa-check"></i><b>5.2</b> Overview of Dimension Reduction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="5-2-overview-of-dimension-reduction.html"><a href="5-2-overview-of-dimension-reduction.html#tours-explore-the-subspace-of-d-dimensional-projections"><i class="fa fa-check"></i><b>5.2.1</b> Tours explore the subspace of <span class="math inline">\(d\)</span>-dimensional projections</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html"><i class="fa fa-check"></i><b>5.3</b> Visual Design</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.3.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.3.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.3.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.3.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html"><i class="fa fa-check"></i><b>5.4</b> Software Infrastructure</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.4.1</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#linking-and-highlighting-views-via-interactions"><i class="fa fa-check"></i><b>5.4.2</b> Linking and highlighting views via interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html"><i class="fa fa-check"></i><b>5.5</b> Case Studies</a><ul>
<li class="chapter" data-level="5.5.1" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-1-exploring-spherical-gaussian-clusters"><i class="fa fa-check"></i><b>5.5.1</b> Case Study 1: Exploring spherical Gaussian clusters</a></li>
<li class="chapter" data-level="5.5.2" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-2-exploring-spherical-gaussian-clusters-with-hierarchical-structure"><i class="fa fa-check"></i><b>5.5.2</b> Case Study 2: Exploring spherical Gaussian clusters with hierarchical structure</a></li>
<li class="chapter" data-level="5.5.3" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-3-exploring-data-with-piecewise-linear-structure"><i class="fa fa-check"></i><b>5.5.3</b> Case Study 3: Exploring data with piecewise linear structure</a></li>
<li class="chapter" data-level="5.5.4" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-4-clustering-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.5.4</b> Case Study 4: Clustering single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="5-6-discussion-3.html"><a href="5-6-discussion-3.html"><i class="fa fa-check"></i><b>5.6</b> Discussion</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-4.html"><a href="acknowledgements-4.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="supplementary-materials.html"><a href="supplementary-materials.html"><i class="fa fa-check"></i>Supplementary Materials</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-software-development.html"><a href="6-1-software-development.html"><i class="fa fa-check"></i><b>6.1</b> Software Development</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-further-work.html"><a href="6-2-further-work.html"><i class="fa fa-check"></i><b>6.2</b> Further Work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="case-studies" class="section level2">
<h2><span class="header-section-number">5.5</span> Case Studies</h2>
<p>The next section steps through case studies cases of our approach using simulated
and real data.</p>
<p>The first three case studies use simulations where the cluster structure and
geometry of the underlying data is known. We start with a simple example
where we generated spherical clusters that are embedded well by t-SNE. Then
we move onto more complex examples where the tour provides insight, such
as clusters that have substructure and where there is more complex geometry
in the data.</p>
<p>In the final case study, we apply our approach to clustering
the mouse retina data from <span class="citation">Macosko et al. (<a href="bibliography.html#ref-Macosko2015-ot">2015</a>)</span>, and apply the tour
to the process of verifying marker genes that separate clusters.</p>
<p>We <em>strongly</em> recommend viewing the linked videos for each case study
while reading. The videos presented show the visual
appearance of the <strong>liminal</strong> interface, and how we can interact with the tour
via the controls previously described.</p>
<div id="case-study-1-exploring-spherical-gaussian-clusters" class="section level3">
<h3><span class="header-section-number">5.5.1</span> Case Study 1: Exploring spherical Gaussian clusters</h3>
<p>To begin we look at simulated datasets that reproduce known facts
about the t-SNE algorithm. Our first data set consists of five spherical 5-<span class="math inline">\(d\)</span>
Gaussian clusters embedded in 10-<span class="math inline">\(d\)</span> space, each cluster has the same
covariance matrix. We then computed a t-SNE layout with default settings
using the <strong>Rtsne</strong> package <span class="citation">(Krijthe <a href="bibliography.html#ref-Rtsne">2015</a>)</span>,
and set up the <strong>liminal</strong> linked interface with grand tour on the 10-<span class="math inline">\(d\)</span>
observations.</p>

<div class="figure" style="text-align: center"><span id="fig:gaussian"></span>
<img src="img/liminal-screenshot-gaussian.png" alt="Screenshots of the liminal interface applied to well clustered data, a video of the tour animation is available at https://player.vimeo.com/video/439635921. (a): t-SNE (left) and tour (right) in side by side layout coloured by cluster. (b): brushing the red cluster on the t-SNE view, shows the corresponding points on the tour." width="100%" />
<p class="caption">
Figure 5.1: Screenshots of the <strong>liminal</strong> interface applied to well clustered data, a video of the tour animation is available at <a href="https://player.vimeo.com/video/439635921" class="uri">https://player.vimeo.com/video/439635921</a>. <em>(a)</em>: t-SNE (left) and tour (right) in side by side layout coloured by cluster. <em>(b)</em>: brushing the red cluster on the t-SNE view, shows the corresponding points on the tour.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635921" width="100%" height="400px">
</iframe>
<p>We see in figure <a href="5-5-case-studies.html#fig:gaussian">5.1</a> and linked video that t-SNE has
correctly split out each cluster and laid them out in a star like formation.
This is agrees with the tour view,
where once we start the animation, the five clusters begin to appear but
generally points are more concentrated in the projection view compared to
the t-SNE layout (figure <a href="5-5-case-studies.html#fig:gaussian">5.1</a>a).
This can be confirmed via brushing the t-SNE view (figure <a href="5-5-case-studies.html#fig:gaussian">5.1</a>b).</p>
</div>
<div id="case-study-2-exploring-spherical-gaussian-clusters-with-hierarchical-structure" class="section level3">
<h3><span class="header-section-number">5.5.2</span> Case Study 2: Exploring spherical Gaussian clusters with hierarchical structure</h3>
<p>Next we view Gaussian clusters from the
<em>Multi Challenge Dataset</em>, a benchmark simulation data set for clustering tasks
<span class="citation">(Rauber <a href="bibliography.html#ref-Rauber2009-vh">2009</a>)</span>. This dataset has two Gaussian clusters with equal covariance
embedded in 10-<span class="math inline">\(d\)</span>, and a third cluster
with hierarchical structure. This cluster has two 3-<span class="math inline">\(d\)</span> clusters embedded in
10-<span class="math inline">\(d\)</span>, where the second cluster is subdivided into three smaller clusters,
that are each equidistant from each other and have the same covariance structure.
Again t-SNE has correctly identified the sub-clusters but their relative
locations to each other is distorted, with the orange and blue groups being
far from each other in the tour view (figure <a href="5-5-case-studies.html#fig:hierarchical">5.2</a>a). We see
in this case that is difficult
to see the sub-clusters in the tour view, however, once we zoom and highlight
they become more apparent (figure <a href="5-5-case-studies.html#fig:hierarchical">5.2</a>b). When we brush
the sub-clusters in the t-SNE, their
relative placement is again exaggerated, with the tour showing that they
are indeed much closer than the impression the t-SNE view gives.</p>

<div class="figure" style="text-align: center"><span id="fig:hierarchical"></span>
<img src="img/liminal-screenshot-hierarchical.png" alt="Screenshots of the liminal interface applied to sub-clustered data, a video of the tour animation is available at https://player.vimeo.com/video/43963590. (a) t-SNE has identified and spread out the clusters with substructure. (b) Using linked highlighting and brushing, the tour view shows that sub-clusters are much closer together in high dimensional space than the t-SNE view implies." width="100%" />
<p class="caption">
Figure 5.2: Screenshots of the <strong>liminal</strong> interface applied to sub-clustered data, a video of the tour animation is available at <a href="https://player.vimeo.com/video/43963590" class="uri">https://player.vimeo.com/video/43963590</a>. <em>(a)</em> t-SNE has identified and spread out the clusters with substructure. <em>(b)</em> Using linked highlighting and brushing, the tour view shows that sub-clusters are much closer together in high dimensional space than the t-SNE view implies.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635905" width="100%" height="400px">
</iframe>
</div>
<div id="case-study-3-exploring-data-with-piecewise-linear-structure" class="section level3">
<h3><span class="header-section-number">5.5.3</span> Case Study 3: Exploring data with piecewise linear structure</h3>
<p>Next explore some simulated noisy tree structured data (figure
<a href="5-5-case-studies.html#fig:fake-trees">5.3</a>), our interest here is how t-SNE visualisations break
topology of the data, and then seeing if we can resolve this by tweaking the
default parameters with reference to the global view of the data set.</p>

<div class="figure" style="text-align: center"><span id="fig:fake-trees"></span>
<img src="img/tree-view.png" alt="Example high-dimensional tree shaped data, \(n = 3000\) and \(p = 100\). (a) The true data lies 2-d tree like structure consisting of ten branches. This data is available in the phateR package and is simulated via diffusion-limited aggregation (a random walk along the branches of the tree) with Gaussian noise added (Moon et al. 2019). (b) The first two principal components, which form the initial projection for the tour, note that the backbone of the tree is obscured by this view. (c) The default t-SNE view breaks the global structure of the tree. (d) Altering t-SNE using the first two principal components as the starting coordinates for the embedding, results in clustering the tree at its branching points." width="100%" />
<p class="caption">
Figure 5.3: Example high-dimensional tree shaped data, <span class="math inline">\(n = 3000\)</span> and <span class="math inline">\(p = 100\)</span>. <em>(a)</em> The true data lies 2-d tree like structure consisting of ten branches. This data is available in the <strong>phateR</strong> package and is simulated via diffusion-limited aggregation (a random walk along the branches of the tree) with Gaussian noise added <span class="citation">(Moon et al. <a href="bibliography.html#ref-Moon2019-ce">2019</a>)</span>. <em>(b)</em> The first two principal components, which form the initial projection for the tour, note that the backbone of the tree is obscured by this view. <em>(c)</em> The default t-SNE view breaks the global structure of the tree. <em>(d)</em> Altering t-SNE using the first two principal components as the starting coordinates for the embedding, results in clustering the tree at its branching points.
</p>
</div>
<p>First, we apply principal components and restrict the results down to the first
twelve principal components (which makes up approximately 70% of the variance
explained in the data) to use with the grand tour.</p>
<p>Moreover, we run t-SNE using the default arguments on the complete data (this
keeps the first 50 PCs, sets the perplexity to equal 30 and performs random
initialisation). We then create a linked
tour with t-SNE layout with <strong>liminal</strong> as shown in figure <a href="5-5-case-studies.html#fig:trees-01">5.4</a>.</p>

<div class="figure" style="text-align: center"><span id="fig:trees-01"></span>
<img src="img/liminal-screenshot-trees-01.png" alt="Screenshots of the liminal interface applied to tree structured data, a video of the tour animation is available at https://player.vimeo.com/video/439635892. (a): Default t-SNE has difficulties embedding the tree. (b) linked highlighting shows that the backbone of the tree has been broken by the embedding." width="100%" />
<p class="caption">
Figure 5.4: Screenshots of the <strong>liminal</strong> interface applied to tree structured data, a video of the tour animation is available at <a href="https://player.vimeo.com/video/439635892" class="uri">https://player.vimeo.com/video/439635892</a>. <em>(a)</em>: Default t-SNE has difficulties embedding the tree. <em>(b)</em> linked highlighting shows that the backbone of the tree has been broken by the embedding.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635892" width="100%" height="400px">
</iframe>
<p>We see that the t-SNE view has been unable to recapitulate the
topology of the tree - the backbone (blue) branch has been split into three
fragments (figure <a href="5-5-case-studies.html#fig:trees-01">5.4</a>a). We can see this immediately via the linked highlighting over both
plots. If we click on the legend for the zero branch, the blue coloured points
on each view are highlighted and the remaining points are made transparent.
From, here it becomes apparent from the tour view that the blue branch forms
the backbone of the tree and is connected to all other branches. We can also
again see that cluster sizes formed via t-SNE can be misleading; from the tour
view there is a lot of noise along the branches, while this does not appear to
be the case for the t-SNE result (figure <a href="5-5-case-studies.html#fig:trees-01">5.4</a>b).</p>
<p>From the first view, we modify the inputs to the t-SNE view, to try and produce
a better trade-off between local structure and retain the topology of the data.
We keep every parameter the same except that we initialise <span class="math inline">\(Y\)</span> with the first
two PCs (scaled to have standard deviation 1e-4) instead of the default random
initialisation and increase the perplexity from 30 to 100. We then combine
these results with our tour view as displayed in figure <a href="5-5-case-studies.html#fig:trees-02">5.5</a>.</p>

<div class="figure" style="text-align: center"><span id="fig:trees-02"></span>
<img src="img/liminal-screenshot-trees-02.png" alt="Screenshots of the liminal interface applied to tree structured data, a video of the tour animation is available at https://player.vimeo.com/video/439635863." width="100%" />
<p class="caption">
Figure 5.5: Screenshots of the <strong>liminal</strong> interface applied to tree structured data, a video of the tour animation is available at <a href="https://player.vimeo.com/video/439635863" class="uri">https://player.vimeo.com/video/439635863</a>.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635863" width="100%" height="400px">
</iframe>
<p>Figure @Screenshots of the <strong>liminal</strong> interface applied to tree structured data, a video of the tour animation is available at <a href="https://player.vimeo.com/video/439635863" class="uri">https://player.vimeo.com/video/439635863</a>. shows that this selection of parameters results in the
tips of the branches (the three black dots in figure <a href="5-5-case-studies.html#fig:fake-trees">5.3</a>a)
being split into three clusters representing the terminal branches of the tree.
However, there are perceptual issues following the placement of the three
groupings on the t-SNE view that become apparent via simple linked brushing.
If we brush the
tips of the yellow and brown branches (which appear to be close to each other
on the t-SNE view), we immediately see the placement is distorted in the t-SNE
view, and in the tour view these tips are at opposite ends of the tree
(figure <a href="5-5-case-studies.html#fig:trees-02">5.5</a>b).
Although, this is a known issue of the t-SNE algorithm, we can easily identify
it via simple interactivity without knowing the inner workings of the method.</p>
</div>
<div id="case-study-4-clustering-single-cell-rna-seq-data" class="section level3">
<h3><span class="header-section-number">5.5.4</span> Case Study 4: Clustering single cell RNA-seq data</h3>
<p>A common analysis task in single cell studies is performing clustering to
identify groupings of cells with similar expression profiles. Analysts
in this area generally use non linear DR methods for verification and identification
of clusters and developmental trajectories (i.e., case study 1). For clustering,
workflows the primary task is verify the existence of clusters and then begin to
identify the clusters as cell types using the expression of “known”
marker genes. Here a ‘faithful’ a embedding should ideally preserve the topology
of the data; cells that correspond to a cell type should lie in the same
neighbourhood in high-dimensional space. In this case study we use our
linked brushing approaches to look at neighbourhood preservation
and look at marker genes through the lens of the tour.</p>
<p>First, we downloaded the raw mouse retinal single cell RNA-seq data from
<span class="citation">Macosko et al. (<a href="bibliography.html#ref-Macosko2015-ot">2015</a>)</span> using the <strong>scRNAseq</strong> Bioconductor package <span class="citation">(Risso and Cole <a href="bibliography.html#ref-scRNAseq-d">2019</a>)</span>. We
have followed a standard workflow for pre-processing and normalizing this data
(described by <span class="citation">Amezquita et al. (<a href="bibliography.html#ref-Amezquita2020-at">2020</a>)</span>): we performed QC using the the <strong>scater</strong>
by removing cells with high proportion of mitochondrial gene expression and low
numbers of genes detected, we log-transformed and normalised the expression
values and finally selected highly variable genes (HVGs) using <strong>scran</strong>
<span class="citation">(McCarthy et al. <a href="bibliography.html#ref-McCarthy2017">2017</a>; Lun, McCarthy, and Marioni <a href="bibliography.html#ref-Lun2016">2016</a>)</span>. The top ten percent of HVGs were used to subset
the normalised expression matrix and compute PCA using the first 25 components.
Using the PCs we built a shared nearest neighbours graph (with <span class="math inline">\(k = 10\)</span>) and
used Louvain clustering to generate clusters <span class="citation">(Blondel et al. <a href="bibliography.html#ref-Blondel2008-bx">2008</a>)</span>.</p>
<p>To check and verify the clustering we construct a <strong>liminal</strong> view.
We tour the first five PCs (approximately 20% of the variance in expression),
along side the t-SNE view which was computed from all 25 PCs.
Due to latency of the interface we do weighted sample of rows based on
cluster membership, leaving us with approximately 10 per cent of the original
data size - 4,590 cells. Although this is not ideal, it still allows us to
get a sense of the shape of the clusters as seen in figure <a href="5-5-case-studies.html#fig:mouse-01">5.6</a>.</p>

<div class="figure" style="text-align: center"><span id="fig:mouse-01"></span>
<img src="img/liminal-screenshot-mouse-01.png" alt="Screenshots of the liminal interface applied to single cell data. A video of the tour animation is available at https://player.vimeo.com/video/439635812. (a): linked brushing identifies distortions in cluster placement by the embedding. (b): t-SNE spreads out larger clusters like cluster 2, 4, 6, and 11 which are concentrated in the tour view. (c): the tour view gives a sense of the geometry of the data; the PCs appear to form a tetrahedron like shape. (d): brushing on the tour view highlights diffusions, points in cluster 9 are close together in the tour view are far away in the t-SNE view." width="100%" />
<p class="caption">
Figure 5.6: Screenshots of the <strong>liminal</strong> interface applied to single cell data. A video of the tour animation is available at <a href="https://player.vimeo.com/video/439635812" class="uri">https://player.vimeo.com/video/439635812</a>. <em>(a)</em>: linked brushing identifies distortions in cluster placement by the embedding. <em>(b)</em>: t-SNE spreads out larger clusters like cluster 2, 4, 6, and 11 which are concentrated in the tour view. <em>(c)</em>: the tour view gives a sense of the geometry of the data; the PCs appear to form a tetrahedron like shape. <em>(d)</em>: brushing on the tour view highlights diffusions, points in cluster 9 are close together in the tour view are far away in the t-SNE view.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635812" width="100%" height="400px">
</iframe>
<p>We observe that the embedding has mostly captured the
clusters relative location to each other to their location in high dimensional space, with a notable exception of points in cluster 3 and 10 as shown with
linked brushing (figure <a href="5-5-case-studies.html#fig:mouse-01">5.6</a>a). As expected, t-SNE mitigates the crowding problem that is
an issue for tour in this case, where points in clusters 2,4,6, and 11 are
clumped together in tour view, but are blown up in the embedding view (figure <a href="5-5-case-studies.html#fig:mouse-01">5.6</a>b).
The tour appears to form a tetrahedron like shape, with points lying on
surface and along the vertices of the the tetrahedron in 5-<span class="math inline">\(d\)</span> PCA space -
a phenomena that has also been observed in <span class="citation">Korem et al. (<a href="bibliography.html#ref-Korem2015-af">2015</a>)</span>
(figure <a href="5-5-case-studies.html#fig:mouse-01">5.6</a>c). Brushing on
the tour view, reveals points in cluster 9 that are diffuse in the embedding
view, points in cluster 9 are relatively far away and spread apart from other
clusters in the tour view, but has points placed in cluster 3 and 9 in the embedding
(figure <a href="5-5-case-studies.html#fig:mouse-01">5.6</a>d).</p>
<p>Next, we identify marker genes for clusters using one sided Welch t-tests with
a minimum log fold change of one as recommended by <span class="citation">Amezquita et al. (<a href="bibliography.html#ref-Amezquita2020-at">2020</a>)</span>, which
uses the testing framework from <span class="citation">McCarthy and Smyth (<a href="bibliography.html#ref-McCarthy2009-qx">2009</a>)</span>. We select the top 10 marker
genes that are upregulated in cluster 2, which was one of the clumped clusters
when we toured on principal components. Here the tour becomes an alternative
to a standard heatmap view for assessing shared markers; the basis
generation (shown as the biplot on the left view) reflects the relative
weighting of each gene. We run the tour directly on the
log normalised expression values using the same subset as before figure <a href="5-5-case-studies.html#fig:mouse-02">5.7</a>.</p>

<div class="figure" style="text-align: center"><span id="fig:mouse-02"></span>
<img src="img/liminal-screenshot-mouse-02.png" alt="Screenshots of the liminal tour applied to a marker gene set. A video of the tour animation is available at https://player.vimeo.com/video/439635843. (a): the axis (left) and tour (right) view of 10 marker genes, using linked highlighting we see that the marker genes separate clusters 2, 4, 6, and 11 from the other groups. (b): after zooming, it appears that the marker genes give rise to a gradient along the clusters." width="100%" />
<p class="caption">
Figure 5.7: Screenshots of the <strong>liminal</strong> tour applied to a marker gene set. A video of the tour animation is available at <a href="https://player.vimeo.com/video/439635843" class="uri">https://player.vimeo.com/video/439635843</a>. <em>(a)</em>: the axis (left) and tour (right) view of 10 marker genes, using linked highlighting we see that the marker genes separate clusters 2, 4, 6, and 11 from the other groups. <em>(b)</em>: after zooming, it appears that the marker genes give rise to a gradient along the clusters.
</p>
</div>
<iframe src="https://player.vimeo.com/video/439635843" width="100%" height="400px">
</iframe>
<p>We see that the expression of the marker genes, appear to separate
the previously clumped clusters 2,4,6, and 11 from the other clusters,
indicating that these genes are expressed in all four clusters (figure <a href="5-5-case-studies.html#fig:mouse-02">5.7</a>a). Using zooming, we can see a trajectory forming along
the clusters, while the axis view shows that magnitude of expression in
the marker genes is similar across these separated clusters which is
consistent with the results of marker gene analysis (figure <a href="5-5-case-studies.html#fig:mouse-02">5.7</a>b).</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="5-4-software-infrastructure.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="5-6-discussion-3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//05-embeddings.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
