<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Bibliography | Fluent statistical computing interfaces for biological data analysis</title>
  <meta name="description" content="Bibliography | Fluent statistical computing interfaces for biological data analysis" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Bibliography | Fluent statistical computing interfaces for biological data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Bibliography | Fluent statistical computing interfaces for biological data analysis" />
  
  
  

<meta name="author" content="Stuart Lee" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="6-2-further-work.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="template/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./"></a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="0-1-open-research.html"><a href="0-1-open-research.html"><i class="fa fa-check"></i><b>0.1</b> Open Research</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="1-ch-intro.html"><a href="1-ch-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-1-sec-tidygranges.html"><a href="1-1-sec-tidygranges.html"><i class="fa fa-check"></i><b>1.1</b> A grammar for genomic data analysis</a></li>
<li class="chapter" data-level="1.2" data-path="1-2-sec-integrate.html"><a href="1-2-sec-integrate.html"><i class="fa fa-check"></i><b>1.2</b> Integration of genomic data structures</a></li>
<li class="chapter" data-level="1.3" data-path="1-3-sec-coverage.html"><a href="1-3-sec-coverage.html"><i class="fa fa-check"></i><b>1.3</b> Representation of genomic data structures</a></li>
<li class="chapter" data-level="1.4" data-path="1-4-sec-va.html"><a href="1-4-sec-va.html"><i class="fa fa-check"></i><b>1.4</b> Interactive visualisation for high-dimensional data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-plyranges.html"><a href="2-ch-plyranges.html"><i class="fa fa-check"></i><b>2</b> plyranges: a grammar of data transformation for genomics</a><ul>
<li class="chapter" data-level="2.1" data-path="2-1-background.html"><a href="2-1-background.html"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="2-2-results.html"><a href="2-2-results.html"><i class="fa fa-check"></i><b>2.2</b> Results</a><ul>
<li class="chapter" data-level="2.2.1" data-path="2-2-results.html"><a href="2-2-results.html#genomic-relational-algebra"><i class="fa fa-check"></i><b>2.2.1</b> Genomic Relational Algebra</a></li>
<li class="chapter" data-level="2.2.2" data-path="2-2-results.html"><a href="2-2-results.html#developing-workflows-with-plyranges"><i class="fa fa-check"></i><b>2.2.2</b> Developing workflows with <code>plyranges</code></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="2-3-discussion.html"><a href="2-3-discussion.html"><i class="fa fa-check"></i><b>2.3</b> Discussion</a></li>
<li class="chapter" data-level="2.4" data-path="2-4-conclusion.html"><a href="2-4-conclusion.html"><i class="fa fa-check"></i><b>2.4</b> Conclusion</a></li>
<li class="chapter" data-level="2.5" data-path="2-5-availability-of-data-and-materials.html"><a href="2-5-availability-of-data-and-materials.html"><i class="fa fa-check"></i><b>2.5</b> Availability of Data and Materials</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-1.html"><a href="acknowledgements-1.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-fluentGenomics.html"><a href="3-ch-fluentGenomics.html"><i class="fa fa-check"></i><b>3</b> Fluent genomics with <em>plyranges</em> and <em>tximeta</em></a><ul>
<li class="chapter" data-level="3.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-1-introduction.html"><a href="3-1-introduction.html#experimental-data"><i class="fa fa-check"></i><b>3.1.1</b> Experimental Data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="3-2-se.html"><a href="3-2-se.html"><i class="fa fa-check"></i><b>3.2</b> Import Data as a <em>SummarizedExperiment</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="3-2-se.html"><a href="3-2-se.html#using-tximeta-to-import-rna-seq-quantification-data"><i class="fa fa-check"></i><b>3.2.1</b> Using <code>tximeta</code> to import RNA-seq quantification data</a></li>
<li class="chapter" data-level="3.2.2" data-path="3-2-se.html"><a href="3-2-se.html#atac"><i class="fa fa-check"></i><b>3.2.2</b> Importing ATAC-seq data as a <em>SummarizedExperiment</em> object</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html"><i class="fa fa-check"></i><b>3.3</b> Model assays</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#rna-seq-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq differential gene expression analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="3-3-model-assays.html"><a href="3-3-model-assays.html#atac-seq-peak-differential-abundance-analysis"><i class="fa fa-check"></i><b>3.3.2</b> ATAC-seq peak differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html"><i class="fa fa-check"></i><b>3.4</b> Integrate ranges</a><ul>
<li class="chapter" data-level="3.4.1" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#finding-overlaps-with-plyranges"><i class="fa fa-check"></i><b>3.4.1</b> Finding overlaps with <code>plyranges</code></a></li>
<li class="chapter" data-level="3.4.2" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#down-sampling-non-differentially-expressed-genes"><i class="fa fa-check"></i><b>3.4.2</b> Down sampling non-differentially expressed genes</a></li>
<li class="chapter" data-level="3.4.3" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#expanding-genomic-coordinates-around-the-transcription-start-site"><i class="fa fa-check"></i><b>3.4.3</b> Expanding genomic coordinates around the transcription start site</a></li>
<li class="chapter" data-level="3.4.4" data-path="3-4-integrate-ranges.html"><a href="3-4-integrate-ranges.html#use-overlap-joins-to-find-relative-enrichment"><i class="fa fa-check"></i><b>3.4.4</b> Use overlap joins to find relative enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="3-5-discussion-1.html"><a href="3-5-discussion-1.html"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="3-6-software-availability.html"><a href="3-6-software-availability.html"><i class="fa fa-check"></i><b>3.6</b> Software Availability</a></li>
<li class="chapter" data-level="" data-path="acknowledgements-2.html"><a href="acknowledgements-2.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-ch-intron.html"><a href="4-ch-intron.html"><i class="fa fa-check"></i><b>4</b> Exploratory coverage analysis with superintronic and plyranges</a><ul>
<li class="chapter" data-level="4.1" data-path="4-1-introduction-1.html"><a href="4-1-introduction-1.html"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-2-methods.html"><a href="4-2-methods.html"><i class="fa fa-check"></i><b>4.2</b> Methods</a><ul>
<li class="chapter" data-level="4.2.1" data-path="4-2-methods.html"><a href="4-2-methods.html#representation-of-coverage-estimation"><i class="fa fa-check"></i><b>4.2.1</b> Representation of coverage estimation</a></li>
<li class="chapter" data-level="4.2.2" data-path="4-2-methods.html"><a href="4-2-methods.html#integration-of-external-annotations"><i class="fa fa-check"></i><b>4.2.2</b> Integration of external annotations</a></li>
<li class="chapter" data-level="4.2.3" data-path="4-2-methods.html"><a href="4-2-methods.html#discovery-of-regions-of-interest-via-data-descriptors"><i class="fa fa-check"></i><b>4.2.3</b> Discovery of regions of interest via ‘data descriptors’</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><a href="4-3-a-workflow-for-uncovering-intron-retention-in-a-zebrafish-experiment.html"><i class="fa fa-check"></i><b>4.3</b> A workflow for uncovering intron retention in a zebrafish experiment</a></li>
<li class="chapter" data-level="4.4" data-path="4-4-discussion-2.html"><a href="4-4-discussion-2.html"><i class="fa fa-check"></i><b>4.4</b> Discussion</a></li>
<li class="chapter" data-level="4.5" data-path="4-5-acknowledgements-3.html"><a href="4-5-acknowledgements-3.html"><i class="fa fa-check"></i><b>4.5</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-ch-tsne.html"><a href="5-ch-tsne.html"><i class="fa fa-check"></i><b>5</b> Casting multiple shadows: high-dimensional interactive data visualisation with tours and embeddings</a><ul>
<li class="chapter" data-level="5.1" data-path="5-1-introduction-2.html"><a href="5-1-introduction-2.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-2-dimensionality-reduction-background.html"><a href="5-2-dimensionality-reduction-background.html"><i class="fa fa-check"></i><b>5.2</b> Dimensionality Reduction Background</a></li>
<li class="chapter" data-level="5.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html"><i class="fa fa-check"></i><b>5.3</b> Visual Design</a><ul>
<li class="chapter" data-level="5.3.1" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#finding-gestalt-focus-and-context"><i class="fa fa-check"></i><b>5.3.1</b> Finding Gestalt: focus and context</a></li>
<li class="chapter" data-level="5.3.2" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#posing-queries-multiple-views-many-contexts"><i class="fa fa-check"></i><b>5.3.2</b> Posing Queries: multiple views, many contexts</a></li>
<li class="chapter" data-level="5.3.3" data-path="5-3-visual-design.html"><a href="5-3-visual-design.html#making-comparisons-revising-embeddings"><i class="fa fa-check"></i><b>5.3.3</b> Making comparisons: revising embeddings</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html"><i class="fa fa-check"></i><b>5.4</b> Software Infrastructure</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-explore-the-subspace-of-d-dimensional-projections"><i class="fa fa-check"></i><b>5.4.1</b> Tours explore the subspace of <span class="math inline">\(d\)</span>-dimensional projections</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#tours-as-a-streaming-data-problem"><i class="fa fa-check"></i><b>5.4.2</b> Tours as a streaming data problem</a></li>
<li class="chapter" data-level="5.4.3" data-path="5-4-software-infrastructure.html"><a href="5-4-software-infrastructure.html#linking-and-highlighting-views-via-interactions"><i class="fa fa-check"></i><b>5.4.3</b> Linking and highlighting views via interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html"><i class="fa fa-check"></i><b>5.5</b> Case Studies</a><ul>
<li class="chapter" data-level="5.5.1" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-1-exploring-simulation-data-with-tours-and-t-sne"><i class="fa fa-check"></i><b>5.5.1</b> Case Study 1: Exploring simulation data with tours and t-SNE</a></li>
<li class="chapter" data-level="5.5.2" data-path="5-5-case-studies.html"><a href="5-5-case-studies.html#case-study-2-clustering-single-cell-rna-seq-data"><i class="fa fa-check"></i><b>5.5.2</b> Case Study 2: Clustering single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="5-6-discussion-3.html"><a href="5-6-discussion-3.html"><i class="fa fa-check"></i><b>5.6</b> Discussion</a></li>
<li class="chapter" data-level="5.7" data-path="5-7-acknowledgements-4.html"><a href="5-7-acknowledgements-4.html"><i class="fa fa-check"></i><b>5.7</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-conclusion.html"><a href="6-ch-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="6-1-software-development.html"><a href="6-1-software-development.html"><i class="fa fa-check"></i><b>6.1</b> Software Development</a></li>
<li class="chapter" data-level="6.2" data-path="6-2-further-work.html"><a href="6-2-further-work.html"><i class="fa fa-check"></i><b>6.2</b> Further Work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><strong><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></strong></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fluent statistical computing interfaces for biological data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bibliography" class="section level1 unnumbered">
<h1>Bibliography</h1>

<div id="refs" class="references">
<div id="ref-alasooZenodo">
<p>Alasoo, Kaur, and Daniel Gaffney. 2017. “Processed read counts from macrophage RNA-seq and ATAC-seq experiments.” Zenodo. <a href="https://doi.org/10.5281/zenodo.1188300">https://doi.org/10.5281/zenodo.1188300</a>.</p>
</div>
<div id="ref-alasoo">
<p>Alasoo, K, J Rodrigues, S Mukhopadhyay, AJ Knights, AL Mann, K Kundu, HIPSCI-Consortium, C Hale, Dougan G, and DJ Gaffney. 2018. “Shared genetic effects on chromatin and gene expression indicate a role for enhancer priming in immune response.” <em>Nature Genetics</em> 50: 424–31. <a href="https://doi.org/10.1038/s41588-018-0046-7">https://doi.org/10.1038/s41588-018-0046-7</a>.</p>
</div>
<div id="ref-Amezquita2020-at">
<p>Amezquita, Robert A, Aaron T L Lun, Etienne Becht, Vince J Carey, Lindsay N Carpp, Ludwig Geistlinger, Federico Marini, et al. 2020. “Orchestrating Single-Cell Analysis with Bioconductor.” <em>Nat. Methods</em> 17 (2): 137–45. <a href="https://doi.org/10.1038/s41592-019-0654-x">https://doi.org/10.1038/s41592-019-0654-x</a>.</p>
</div>
<div id="ref-R-magrittr">
<p>Bache, Stefan Milton, and Hadley Wickham. 2014. <em>Magrittr: A Forward-Pipe Operator for R</em>. <a href="https://CRAN.R-project.org/package=magrittr">https://CRAN.R-project.org/package=magrittr</a>.</p>
</div>
<div id="ref-ncbi-geo">
<p>Barrett, Tanya, Stephen E. Wilhite, Pierre Ledoux, Carlos Evangelista, Irene F. Kim, Maxim Tomashevsky, Kimberly A. Marshall, et al. 2013. “NCBI Geo: Archive for Functional Genomics Data Sets—Update.” <em>Nucleic Acids Research</em> 41 (D1): D991–D995. <a href="https://doi.org/10.1093/nar/gks1193">https://doi.org/10.1093/nar/gks1193</a>.</p>
</div>
<div id="ref-fnn-pkg">
<p>Beygelzimer, Alina, Sham Kakadet, John Langford, Sunil Arya, David Mount, and Shengqiao Li. 2019. <em>FNN: Fast Nearest Neighbor Search Algorithms and Applications</em>. <a href="https://CRAN.R-project.org/package=FNN">https://CRAN.R-project.org/package=FNN</a>.</p>
</div>
<div id="ref-Blondel2008-bx">
<p>Blondel, Vincent D, Jean-Loup Guillaume, Renaud Lambiotte, and Etienne Lefebvre. 2008. “Fast Unfolding of Communities in Large Networks.” <em>J. Stat. Mech.</em> 2008 (10). IOP Publishing: P10008. <a href="https://doi.org/10.1088/1742-5468/2008/10/P10008">https://doi.org/10.1088/1742-5468/2008/10/P10008</a>.</p>
</div>
<div id="ref-Brehmer2014-hk">
<p>Brehmer, Matthew, Michael Sedlmair, Stephen Ingram, and Tamara Munzner. 2014. “Visualizing Dimensionally-Reduced Data: Interviews with Analysts and a Characterization of Task Sequences.” In <em>Proceedings of the Fifth Workshop on Beyond Time and Errors: Novel Evaluation Methods for Visualization</em>, 1–8. dl.acm.org. <a href="https://dl.acm.org/doi/abs/10.1145/2669557.2669559">https://dl.acm.org/doi/abs/10.1145/2669557.2669559</a>.</p>
</div>
<div id="ref-Buja1996-fk">
<p>Buja, Andreas, Dianne Cook, and Deborah F Swayne. 1996. “Interactive High-Dimensional Data Visualization.” <em>J. Comput. Graph. Stat.</em> 5 (1). [American Statistical Association, Taylor &amp; Francis, Ltd., Institute of Mathematical Statistics, Interface Foundation of America]: 78–99. <a href="https://doi.org/10.2307/1390754">https://doi.org/10.2307/1390754</a>.</p>
</div>
<div id="ref-Chang2020-bq">
<p>Chang, Winston, Joe Cheng, J J Allaire, Yihui Xie, and Jonathan McPherson. 2020. “Shiny: Web Application Framework for R.” <a href="https://CRAN.R-project.org/package=shiny">https://CRAN.R-project.org/package=shiny</a>.</p>
</div>
<div id="ref-Coenen2019-to">
<p>Coenen, Andy, and Adam Pearce. 2019. “Understanding UMAP.” <a href="https://pair-code.github.io/understanding-umap/">https://pair-code.github.io/understanding-umap/</a>. <a href="https://pair-code.github.io/understanding-umap/">https://pair-code.github.io/understanding-umap/</a>.</p>
</div>
<div id="ref-Coifman2005-ak">
<p>Coifman, R R, S Lafon, A B Lee, M Maggioni, B Nadler, F Warner, and S W Zucker. 2005. “Geometric Diffusions as a Tool for Harmonic Analysis and Structure Definition of Data: Diffusion Maps.” <em>Proc. Natl. Acad. Sci. U. S. A.</em> 102 (21): 7426–31. <a href="https://doi.org/10.1073/pnas.0500334102">https://doi.org/10.1073/pnas.0500334102</a>.</p>
</div>
<div id="ref-Cook1995-bi">
<p>Cook, Dianne, Andreas Buja, Javier Cabrera, and Catherine Hurley. 1995. “Grand Tour and Projection Pursuit.” <em>J. Comput. Graph. Stat.</em> 4 (3). Taylor &amp; Francis: 155–72. <a href="https://doi.org/10.1080/10618600.1995.10474674">https://doi.org/10.1080/10618600.1995.10474674</a>.</p>
</div>
<div id="ref-Dale2011-js">
<p>Dale, Ryan K, Brent S Pedersen, and Aaron R Quinlan. 2011. “Pybedtools: A Flexible Python Library for Manipulating Genomic Datasets and Annotations.” <em>Bioinformatics</em> 27 (24): 3423–4. <a href="https://doi.org/10.1093/bioinformatics/btr539">https://doi.org/10.1093/bioinformatics/btr539</a>.</p>
</div>
<div id="ref-rcpp-annoy-pkg">
<p>Eddelbuettel, Dirk. 2020. <em>RcppAnnoy: ’Rcpp’ Bindings for ’Annoy’, a Library for Approximate Nearest Neighbors</em>. <a href="https://CRAN.R-project.org/package=RcppAnnoy">https://CRAN.R-project.org/package=RcppAnnoy</a>.</p>
</div>
<div id="ref-FowlerFluent">
<p>Fowler, Martin. n.d. “FluentInterface.” <a href="https://martinfowler.com/bliki/FluentInterface.html">https://martinfowler.com/bliki/FluentInterface.html</a>. <a href="https://martinfowler.com/bliki/FluentInterface.html">https://martinfowler.com/bliki/FluentInterface.html</a>.</p>
</div>
<div id="ref-gencode">
<p>Frankish, A, GENCODE-consoritum, and P Flicek. 2018. “GENCODE reference annotation for the human and mouse genomes.” <em>Nucleic Acids Research</em>. <a href="https://doi.org/10.1093/nar/gky955">https://doi.org/10.1093/nar/gky955</a>.</p>
</div>
<div id="ref-Friedman2002-yx">
<p>Friedman, Jerome H, and Werner Stuetzle. 2002. “John W. Tukey’s Work on Interactive Graphics.” <em>Ann. Stat.</em> 30 (6). Institute of Mathematical Statistics: 1629–39. <a href="https://doi.org/10.1214/aos/1043351250">https://doi.org/10.1214/aos/1043351250</a>.</p>
</div>
<div id="ref-Genovese2017-iq">
<p>Genovese, Christopher, Marco Perone-Pacifico, Isabella Verdinelli, and Larry Wasserman. 2017. “Finding Singular Features.” <em>J. Comput. Graph. Stat.</em> 26 (3). Taylor &amp; Francis: 598–609. <a href="https://doi.org/10.1080/10618600.2016.1260472">https://doi.org/10.1080/10618600.2016.1260472</a>.</p>
</div>
<div id="ref-Grolemund2017-uc">
<p>Grolemund, Garrett, and Hadley Wickham. 2017. <em>R for Data Science</em>. <a href="http://r4ds.had.co.nz/">http://r4ds.had.co.nz/</a>.</p>
</div>
<div id="ref-Hansen2012">
<p>Hansen, Kasper D., Rafael A. Irizarry, and Zhijin Wu. 2012. “Removing technical variability in RNA-seq data using condition quantile normalization.” <em>Biostatistics</em> 13 (2): 204–16. <a href="https://doi.org/10.1093/biostatistics/kxr054">https://doi.org/10.1093/biostatistics/kxr054</a>.</p>
</div>
<div id="ref-R-rlang">
<p>Henry, Lionel, and Hadley Wickham. 2017. <em>Rlang: Functions for Base Types and Core R and ’Tidyverse’ Features</em>. <a href="http://rlang.tidyverse.org">http://rlang.tidyverse.org</a>.</p>
</div>
<div id="ref-Huber2015-ei">
<p>Huber, Wolfgang, Vincent J Carey, Robert Gentleman, Simon Anders, Marc Carlson, Benilton S Carvalho, Hector Corrada Bravo, et al. 2015a. “Orchestrating High-Throughput Genomic Analysis with Bioconductor.” <em>Nat. Methods</em> 12 (2). Springer Nature: 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.</p>
</div>
<div id="ref-bioc">
<p>———. 2015b. “Orchestrating High-Throughput Genomic Analysis with Bioconductor.” <em>Nature Methods</em> 12 (2). Springer Nature: 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.</p>
</div>
<div id="ref-Hyndman2015-sc">
<p>Hyndman, R J, E Wang, and N Laptev. 2015. “Large-Scale Unusual Time Series Detection.” In <em>2015 IEEE International Conference on Data Mining Workshop (ICDMW)</em>, 1616–9. <a href="https://doi.org/10.1109/ICDMW.2015.104">https://doi.org/10.1109/ICDMW.2015.104</a>.</p>
</div>
<div id="ref-Kaitoua2017-pw">
<p>Kaitoua, A, Pinoli, P, Bertoni, M, and Ceri, S. 2017. “Framework for Supporting Genomic Operations.” <em>IEEE Trans. Comput.</em> 66 (3): 443–57. <a href="https://doi.org/10.1109/TC.2016.2603980">https://doi.org/10.1109/TC.2016.2603980</a>.</p>
</div>
<div id="ref-Kobak2019-lm">
<p>Kobak, Dmitry, and Philipp Berens. 2019. “The Art of Using t-SNE for Single-Cell Transcriptomics.” <em>Nat. Commun.</em> 10 (1): 5416. <a href="https://doi.org/10.1038/s41467-019-13056-x">https://doi.org/10.1038/s41467-019-13056-x</a>.</p>
</div>
<div id="ref-Korem2015-af">
<p>Korem, Yael, Pablo Szekely, Yuval Hart, Hila Sheftel, Jean Hausser, Avi Mayo, Michael E Rothenberg, Tomer Kalisky, and Uri Alon. 2015. “Geometry of the Gene Expression Space of Individual Cells.” <em>PLoS Comput. Biol.</em> 11 (7): e1004224. <a href="https://doi.org/10.1371/journal.pcbi.1004224">https://doi.org/10.1371/journal.pcbi.1004224</a>.</p>
</div>
<div id="ref-Kozanitis2014-va">
<p>Kozanitis, Christos, Heiberg, Andrew, Varghese, George, and Bafna, Vineet. 2014. “Using Genome Query Language to Uncover Genetic Variation.” <em>Bioinformatics</em> 30 (1): 1–8. <a href="https://doi.org/10.1093/bioinformatics/btt250">https://doi.org/10.1093/bioinformatics/btt250</a>.</p>
</div>
<div id="ref-Kozanitis2016-bm">
<p>Kozanitis, Christos, and David A Patterson. 2016. “GenAp: A Distributed SQL Interface for Genomic Data.” <em>BMC Bioinformatics</em> 17 (February): 63. <a href="https://doi.org/10.1186/s12859-016-0904-1">https://doi.org/10.1186/s12859-016-0904-1</a>.</p>
</div>
<div id="ref-snakemake">
<p>Köster, Johannes, and Sven Rahmann. 2012. “Snakemake—a scalable bioinformatics workflow engine.” <em>Bioinformatics</em> 28 (19): 2520–2. <a href="https://doi.org/10.1093/bioinformatics/bts480">https://doi.org/10.1093/bioinformatics/bts480</a>.</p>
</div>
<div id="ref-Rtsne">
<p>Krijthe, Jesse H. 2015. <em>Rtsne: T-Distributed Stochastic Neighbor Embedding Using Barnes-Hut Implementation</em>. <a href="https://github.com/jkrijthe/Rtsne">https://github.com/jkrijthe/Rtsne</a>.</p>
</div>
<div id="ref-Kruskal1964-cz">
<p>Kruskal, J B. 1964a. “Multidimensional Scaling by Optimizing Goodness of Fit to a Nonmetric Hypothesis.” <em>Psychometrika</em> 29 (1). Springer-Verlag: 1–27. <a href="https://doi.org/10.1007/BF02289565">https://doi.org/10.1007/BF02289565</a>.</p>
</div>
<div id="ref-Kruskal1964-cw">
<p>———. 1964b. “Nonmetric Multidimensional Scaling: A Numerical Method.” <em>Psychometrika</em> 29 (2): 115–29. <a href="https://doi.org/10.1007/BF02289694">https://doi.org/10.1007/BF02289694</a>.</p>
</div>
<div id="ref-Laa2020-wr">
<p>Laa, Ursula, Dianne Cook, Andreas Buja, and German Valencia. 2020. “Hole or Grain? A Section Pursuit Index for Finding Hidden Structure in Multiple Dimensions,” April. <a href="http://arxiv.org/abs/2004.13327">http://arxiv.org/abs/2004.13327</a>.</p>
</div>
<div id="ref-Laa2020-fy">
<p>Laa, Ursula, Dianne Cook, and German Valencia. 2020. “A Slice Tour for Finding Hollowness in High-Dimensional Data.” <em>J. Comput. Graph. Stat.</em>, June. Taylor &amp; Francis, 1–10. <a href="https://doi.org/10.1080/10618600.2020.1777140">https://doi.org/10.1080/10618600.2020.1777140</a>.</p>
</div>
<div id="ref-Law2018-f1000">
<p>Law, Charity W, Monther Alhamdoosh, Shian Su, Xueyi Dong, Luyi Tian, Gordon K Smyth, and Matthew E Ritchie. 2018. “RNA-seq Analysis Is Easy as 1-2-3 with Limma, Glimma and edgeR.” <em>F1000 Research</em> 5 (1408). F1000 Research Limited: 1408. <a href="https://doi.org/10.12688/f1000research.9005.3">https://doi.org/10.12688/f1000research.9005.3</a>.</p>
</div>
<div id="ref-Lawrence2013-wg">
<p>Lawrence, Michael, Wolfgang Huber, Hervé Pagès, Patrick Aboyoun, Marc Carlson, Robert Gentleman, Martin Morgan, and Vincent Carey. 2013a. “Software for Computing and Annotating Genomic Ranges.” <em>PLoS Comput. Biol.</em> 9. <a href="https://doi.org/10.1371/journal.pcbi.1003118">https://doi.org/10.1371/journal.pcbi.1003118</a>.</p>
</div>
<div id="ref-granges">
<p>———. 2013b. “Software for Computing and Annotating Genomic Ranges.” <em>PLoS Comput. Biol.</em> 9. <a href="https://doi.org/10.1371/journal.pcbi.1003118">https://doi.org/10.1371/journal.pcbi.1003118</a>.</p>
</div>
<div id="ref-Lee2009-zb">
<p>Lee, John A, and Michel Verleysen. 2009. “Quality Assessment of Dimensionality Reduction: Rank-Based Criteria.” <em>Neurocomputing</em> 72 (7). Elsevier: 1431–43. <a href="https://doi.org/10.1016/j.neucom.2008.12.017">https://doi.org/10.1016/j.neucom.2008.12.017</a>.</p>
</div>
<div id="ref-Lee2019">
<p>Lee, Stuart, Dianne Cook, and Michael Lawrence. 2019. “Plyranges: A Grammar of Genomic Data Transformation.” <em>Genome Biology</em> 20 (1): 4. <a href="https://doi.org/10.1186/s13059-018-1597-8">https://doi.org/10.1186/s13059-018-1597-8</a>.</p>
</div>
<div id="ref-R-plyranges-z">
<p>Lee, Stuart, Michael Lawrence, and Dianne Cook. 2018. “plyranges: a grammar of genomic data transformation.” <a href="https://doi.org/10.5281/zenodo.1469841">https://doi.org/10.5281/zenodo.1469841</a>.</p>
</div>
<div id="ref-Lee2020-ie">
<p>Lee, Stuart, Michael Lawrence, and Michael I Love. 2020. “Fluent Genomics with <em>Plyranges</em> and <em>Tximeta</em>.” <em>F1000Res.</em> 9 (109). F1000 Research Limited: 109. <a href="https://doi.org/10.12688/f1000research.22259.1">https://doi.org/10.12688/f1000research.22259.1</a>.</p>
</div>
<div id="ref-Lee2019-mf">
<p>Lee, Stuart, Albert Y Zhang, Shian Su, Ashley P Ng, Aliaksei Z Holik, Marie-Liesse Asselin-Labat, Matthew E Ritchie, and Charity W Law. 2019. “Covering All Your Bases: Incorporating Intron Signal from RNA-seq Data.” <em>bioRxiv</em>. <a href="https://doi.org/10.1101/352823">https://doi.org/10.1101/352823</a>.</p>
</div>
<div id="ref-Lewis2012-ai">
<p>Lewis, Joshua, Laurens Van der Maaten, and Virginia de Sa. 2012. “A Behavioral Investigation of Dimensionality Reduction.” In <em>Proceedings of the Annual Meeting of the Cognitive Science Society</em>. Vol. 34. escholarship.org. <a href="https://escholarship.org/content/qt0ss3m55c/qt0ss3m55c.pdf">https://escholarship.org/content/qt0ss3m55c/qt0ss3m55c.pdf</a>.</p>
</div>
<div id="ref-Lex2014-bh">
<p>Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. “UpSet: Visualization of Intersecting Sets.” <em>IEEE Trans. Vis. Comput. Graph.</em> 20 (12): 1983–92. <a href="https://doi.org/10.1109/TVCG.2014.2346248">https://doi.org/10.1109/TVCG.2014.2346248</a>.</p>
</div>
<div id="ref-Li2020-rh">
<p>Li, Mingwei, Zhenge Zhao, and Carlos Scheidegger. 2020. “Visualizing Neural Networks with the Grand Tour.” <em>Distill</em> 5 (3). <a href="https://doi.org/10.23915/distill.00025">https://doi.org/10.23915/distill.00025</a>.</p>
</div>
<div id="ref-Liao2013-eb">
<p>Liao, Yang, Gordon K Smyth, and Wei Shi. 2013. “The Subread Aligner: Fast, Accurate and Scalable Read Mapping by Seed-and-Vote.” <em>Nucleic Acids Res.</em> 41 (10): e108. <a href="https://doi.org/10.1093/nar/gkt214">https://doi.org/10.1093/nar/gkt214</a>.</p>
</div>
<div id="ref-Liao2019-qn">
<p>———. 2019. “The R Package Rsubread Is Easier, Faster, Cheaper and Better for Alignment and Quantification of RNA Sequencing Reads.” <em>Nucleic Acids Res.</em> 47 (8): e47. <a href="https://doi.org/10.1093/nar/gkz114">https://doi.org/10.1093/nar/gkz114</a>.</p>
</div>
<div id="ref-Linderman2019-dq">
<p>Linderman, George C, and Stefan Steinerberger. 2019. “Clustering with t-SNE, Provably.” <em>SIAM Journal on Mathematics of Data Science</em> 1 (2). Society for Industrial; Applied Mathematics: 313–32. <a href="https://doi.org/10.1137/18M1216134">https://doi.org/10.1137/18M1216134</a>.</p>
</div>
<div id="ref-Love2016-f1000">
<p>Love, Michael I, Simon Anders, Vladislav Kim, and Wolfgang Huber. 2016. “RNA-Seq Workflow: Gene-Level Exploratory Analysis and Differential Expression.” <em>F1000 Research</em> 4 (1070). F1000 Research Limited: 1070. <a href="https://doi.org/10.12688/f1000research.7035.2">https://doi.org/10.12688/f1000research.7035.2</a>.</p>
</div>
<div id="ref-Love2014">
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.” <em>Genome Biology</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.</p>
</div>
<div id="ref-Love2019-tximeta">
<p>Love, Michael I, Charlotte Soneson, Peter F Hickey, Lisa K Johnson, N Tessa Pierce, Lori Shepherd, Martin Morgan, and Rob Patro. 2019. “Tximeta: Reference Sequence Checksums for Provenance Identification in RNA-seq.” <em>bioRxiv</em>, September, 777888. <a href="https://doi.org/10.1101/777888">https://doi.org/10.1101/777888</a>.</p>
</div>
<div id="ref-Lun2016">
<p>Lun, Aaron T. L., Davis J. McCarthy, and John C. Marioni. 2016. “A Step-by-Step Workflow for Low-Level Analysis of Single-Cell Rna-Seq Data with Bioconductor.” <em>F1000Res.</em> 5: 2122. <a href="https://doi.org/10.12688/f1000research.9501.2">https://doi.org/10.12688/f1000research.9501.2</a>.</p>
</div>
<div id="ref-Lyttle2020-hp">
<p>Lyttle, Ian, and Vega/Vega-Lite Developers. 2020. “Vegawidget: ’Htmlwidget’ for ’Vega’ and ’Vega-Lite’.” <a href="https://CRAN.R-project.org/package=vegawidget">https://CRAN.R-project.org/package=vegawidget</a>.</p>
</div>
<div id="ref-Macosko2015-ot">
<p>Macosko, Evan Z, Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. “Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.” <em>Cell</em> 161 (5): 1202–14. <a href="https://doi.org/10.1016/j.cell.2015.05.002">https://doi.org/10.1016/j.cell.2015.05.002</a>.</p>
</div>
<div id="ref-Markmiller2014-kg">
<p>Markmiller, Sebastian, Nicole Cloonan, Rea M Lardelli, Karen Doggett, Maria-Cristina Keightley, Yeliz Boglev, Andrew J Trotter, et al. 2014. “Minor Class Splicing Shapes the Zebrafish Transcriptome During Development.” <em>Proc. Natl. Acad. Sci. U. S. A.</em> 111 (8): 3062–7. <a href="https://doi.org/10.1073/pnas.1305536111">https://doi.org/10.1073/pnas.1305536111</a>.</p>
</div>
<div id="ref-McCarthy2017">
<p>McCarthy, Davis J., Kieran R. Campbell, Aaron T. L. Lun, and Quin F. Willis. 2017. “Scater: Pre-Processing, Quality Control, Normalisation and Visualisation of Single-Cell RNA-Seq Data in R.” <em>Bioinformatics</em> 33 (8): 1179–86. <a href="https://doi.org/10.1093/bioinformatics/btw777">https://doi.org/10.1093/bioinformatics/btw777</a>.</p>
</div>
<div id="ref-McCarthy2009-qx">
<p>McCarthy, Davis J, and Gordon K Smyth. 2009. “Testing Significance Relative to a Fold-Change Threshold Is a TREAT.” <em>Bioinformatics</em> 25 (6): 765–71. <a href="https://doi.org/10.1093/bioinformatics/btp053">https://doi.org/10.1093/bioinformatics/btp053</a>.</p>
</div>
<div id="ref-McInnes2018-co">
<p>McInnes, Leland, John Healy, and James Melville. 2018. “UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction,” February. <a href="http://arxiv.org/abs/1802.03426">http://arxiv.org/abs/1802.03426</a>.</p>
</div>
<div id="ref-Melville2020">
<p>Melville, James. n.d. “t-SNE Initialization Options.” <a href="https://jlmelville.github.io/smallvis/init.html">https://jlmelville.github.io/smallvis/init.html</a>. <a href="https://jlmelville.github.io/smallvis/init.html">https://jlmelville.github.io/smallvis/init.html</a>.</p>
</div>
<div id="ref-Meng2016-pz">
<p>Meng, Chen, Oana A Zeleznik, Gerhard G Thallinger, Bernhard Kuster, Amin M Gholami, and Aedı'n C Culhane. 2016. “Dimension Reduction Techniques for the Integrative Analysis of Multi-Omics Data.” <em>Brief. Bioinform.</em> 17 (4): 628–41. <a href="https://doi.org/10.1093/bib/bbv108">https://doi.org/10.1093/bib/bbv108</a>.</p>
</div>
<div id="ref-Moon2019-ce">
<p>Moon, Kevin R, David van Dijk, Zheng Wang, Scott Gigante, Daniel B Burkhardt, William S Chen, Kristina Yim, et al. 2019. “Visualizing Structure and Transitions for Biological Data Exploration.” <em>bioRxiv</em>. <a href="https://doi.org/10.1101/120378">https://doi.org/10.1101/120378</a>.</p>
</div>
<div id="ref-R-ahub">
<p>Morgan, Martin. 2017. <em>AnnotationHub: Client to Access Annotationhub Resources</em>. <a href="https://doi.org/10.18129/B9.bioc.AnnotationHub">https://doi.org/10.18129/B9.bioc.AnnotationHub</a>.</p>
</div>
<div id="ref-rsamtools">
<p>Morgan, Martin, Hervé Pagès, Valerie Obenchain, and Nathaniel Hayden. 2020. <em>Rsamtools: Binary Alignment (BAM), FASTA, Variant Call (BCF), and Tabix File Import</em>. <a href="http://bioconductor.org/packages/Rsamtools">http://bioconductor.org/packages/Rsamtools</a>.</p>
</div>
<div id="ref-Nguyen2019-yh">
<p>Nguyen, Lan Huong, and Susan Holmes. 2019. “Ten Quick Tips for Effective Dimensionality Reduction.” <em>PLoS Comput. Biol.</em> 15 (6): e1006907. <a href="https://doi.org/10.1371/journal.pcbi.1006907">https://doi.org/10.1371/journal.pcbi.1006907</a>.</p>
</div>
<div id="ref-Ovchinnikova2020-sy">
<p>Ovchinnikova, Svetlana, and Simon Anders. 2020. “Exploring Dimension-Reduced Embeddings with Sleepwalk.” <em>Genome Res.</em> 30 (5): 749–56. <a href="https://doi.org/10.1101/gr.251447.119">https://doi.org/10.1101/gr.251447.119</a>.</p>
</div>
<div id="ref-R-biostrings">
<p>Pagès, H., P. Aboyoun, R. Gentleman, and S. DebRoy. 2018. <em>Biostrings: Efficient Manipulation of Biological Strings</em>. <a href="https://doi.org/10.18129/B9.bioc.Biostrings">https://doi.org/10.18129/B9.bioc.Biostrings</a>.</p>
</div>
<div id="ref-salmon">
<p>Patro, R, G Duggal, MI Love, RA Irizarry, and C Kingsford. 2017. “Salmon Provides Fast and Bias-Aware Quantification of Transcript Expression.” <em>Nature Methods</em> 14: 417–19. <a href="https://doi.org/10.1038/nmeth.4197">https://doi.org/10.1038/nmeth.4197</a>.</p>
</div>
<div id="ref-Pezzotti2017-cz">
<p>Pezzotti, Nicola, Boudewijn P F Lelieveldt, Laurens Van Der Maaten, Thomas Hollt, Elmar Eisemann, and Anna Vilanova. 2017. “Approximated and User Steerable tSNE for Progressive Visual Analytics.” <em>IEEE Trans. Vis. Comput. Graph.</em> 23 (7): 1739–52. <a href="https://doi.org/10.1109/TVCG.2016.2570755">https://doi.org/10.1109/TVCG.2016.2570755</a>.</p>
</div>
<div id="ref-Quinlan2010-gc">
<p>Quinlan, Aaron R, and Ira M Hall. 2010. “BEDTools: A Flexible Suite of Utilities for Comparing Genomic Features.” <em>Bioinformatics</em> 26 (6): 841–42. <a href="https://doi.org/10.1093/bioinformatics/btq033">https://doi.org/10.1093/bioinformatics/btq033</a>.</p>
</div>
<div id="ref-Ramirez2014-sh">
<p>Ramı'rez, Fidel, Friederike Dündar, Sarah Diehl, Björn A Grüning, and Thomas Manke. 2014. “DeepTools: A Flexible Platform for Exploring Deep-Sequencing Data.” <em>Nucleic Acids Res.</em> 42 (Web Server issue): W187–91. <a href="https://doi.org/10.1093/nar/gku365">https://doi.org/10.1093/nar/gku365</a>.</p>
</div>
<div id="ref-r-core">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-baser">
<p>———. 2019a. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-baseR">
<p>———. 2019b. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-Rieck2017-kk">
<p>Rieck, Bastian. 2017. “Persistent Homology in Multivariate Data Visualization.” Edited by Heike Little and Michael Gertz. PhD thesis, Ruprecht-Karls-Universität Heidelberg.</p>
</div>
<div id="ref-Kent2017">
<p>Riemondy, Kent A., Ryan M. Sheridan, Austin Gillen, Yinni Yu, Christopher G. Bennett, and Jay R. Hesselberth. 2017. “Valr: Reproducible Genome Interval Arithmetic in R.” <em>F1000Research</em>. <a href="https://doi.org/10.12688/f1000research.11997.1">https://doi.org/10.12688/f1000research.11997.1</a>.</p>
</div>
<div id="ref-scRNAseq-d">
<p>Risso, Davide, and Michael Cole. 2019. <em>ScRNAseq: Collection of Public Single-Cell Rna-Seq Datasets</em>.</p>
</div>
<div id="ref-Ritchie2015-mj">
<p>Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. “Limma Powers Differential Expression Analyses for RNA-sequencing and Microarray Studies.” <em>Nucleic Acids Res.</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.</p>
</div>
<div id="ref-Roadmap-Epigenomics-Consortium2015-pr">
<p>Roadmap Epigenomics Consortium, Anshul Kundaje, Wouter Meuleman, Jason Ernst, Misha Bilenky, Angela Yen, Alireza Heravi-Moussavi, et al. 2015. “Integrative Analysis of 111 Reference Human Epigenomes.” <em>Nature</em> 518 (7539). <a href="https://doi.org/10.1038/nature14248">https://doi.org/10.1038/nature14248</a>.</p>
</div>
<div id="ref-Satyanarayan2017-gs">
<p>Satyanarayan, Arvind, Dominik Moritz, Kanit Wongsuphasawat, and Jeffrey Heer. 2017. “Vega-Lite: A Grammar of Interactive Graphics.” <em>IEEE Trans. Vis. Comput. Graph.</em> 23 (1): 341–50. <a href="https://doi.org/10.1109/TVCG.2016.2599030">https://doi.org/10.1109/TVCG.2016.2599030</a>.</p>
</div>
<div id="ref-ggally">
<p>Schloerke, Barret, Jason Crowley, Di Cook, Francois Briatte, Moritz Marbach, Edwin Thoen, Amos Elberg, and Joseph Larmarange. 2020. <em>GGally: Extension to ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=GGally">https://CRAN.R-project.org/package=GGally</a>.</p>
</div>
<div id="ref-Sedlmair2013-pn">
<p>Sedlmair, Michael, Tamara Munzner, and Melanie Tory. 2013. “Empirical Guidance on Scatterplot and Dimension Reduction Technique Choices.” <em>IEEE Trans. Vis. Comput. Graph.</em> 19 (12): 2634–43. <a href="https://doi.org/10.1109/TVCG.2013.153">https://doi.org/10.1109/TVCG.2013.153</a>.</p>
</div>
<div id="ref-bcfilecache">
<p>Shepherd, Lori, and Martin Morgan. 2019. <em>BiocFileCache: Manage Files Across Sessions</em>.</p>
</div>
<div id="ref-Silva2003-xw">
<p>Silva, V D, and J B Tenenbaum. 2003. “Global Versus Local Methods in Nonlinear Dimensionality Reduction.” <em>Adv. Neural Inf. Process. Syst.</em> papers.nips.cc. <a href="http://papers.nips.cc/paper/2141-global-versus-local-methods-in-nonlinear-dimensionality-reduction.pdf">http://papers.nips.cc/paper/2141-global-versus-local-methods-in-nonlinear-dimensionality-reduction.pdf</a>.</p>
</div>
<div id="ref-Sims2014-qk">
<p>Sims, David, Ian Sudbery, Nicholas E Ilott, Andreas Heger, and Chris P Ponting. 2014. “Sequencing Depth and Coverage: Key Considerations in Genomic Analyses.” <em>Nat. Rev. Genet.</em> 15 (2): 121–32. <a href="https://doi.org/10.1038/nrg3642">https://doi.org/10.1038/nrg3642</a>.</p>
</div>
<div id="ref-Smilkov2016-hp">
<p>Smilkov, Daniel, Nikhil Thorat, Charles Nicholson, Emily Reif, Fernanda B Viégas, and Martin Wattenberg. 2016. “Embedding Projector: Interactive Visualization and Interpretation of Embeddings,” November. <a href="http://arxiv.org/abs/1611.05469">http://arxiv.org/abs/1611.05469</a>.</p>
</div>
<div id="ref-Smyth2004">
<p>Smyth, Gordon K. 2004. “Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments.” <em>Statistical Applications in Genetics and Molecular Biology</em> 3 (1).</p>
</div>
<div id="ref-Soneson2015">
<p>Soneson, Charlotte, Michael I. Love, and Mark Robinson. 2015. “Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences.” <em>F1000Research</em> 4 (1521). <a href="https://doi.org/10.12688/f1000research.7563.1">https://doi.org/10.12688/f1000research.7563.1</a>.</p>
</div>
<div id="ref-Tang2016-oz">
<p>Tang, Jian, Jingzhou Liu, Ming Zhang, and Qiaozhu Mei. 2016. “Visualizing Large-Scale and High-Dimensional Data,” February. <a href="http://arxiv.org/abs/1602.00370">http://arxiv.org/abs/1602.00370</a>.</p>
</div>
<div id="ref-Torgerson1952-am">
<p>Torgerson, Warren S. 1952. “Multidimensional Scaling: I. Theory and Method.” <em>Psychometrika</em> 17 (4). Springer-Verlag: 401–19. <a href="https://doi.org/10.1007/BF02288916">https://doi.org/10.1007/BF02288916</a>.</p>
</div>
<div id="ref-Green1996-qg">
<p>T R G Green AND Petre, M. 1996. “Usability Analysis of Visual Programming Environments: A ‘Cognitive Dimensions’ Framework.” <em>Journal of Visual Languages &amp; Computing</em> 7 (2): 131–74. <a href="https://doi.org/10.1006/jvlc.1996.0009">https://doi.org/10.1006/jvlc.1996.0009</a>.</p>
</div>
<div id="ref-Tukey1977-pn">
<p>Tukey, John W. 1977. <em>Exploratory Data Analysis</em>. Vol. 2. Reading, Mass.: Addison-Wesley Pub. Co.</p>
</div>
<div id="ref-Unwin2018-zb">
<p>Unwin, Antony, and Pedro Valero-Mora. 2018. “Ensemble Graphics.” <em>J. Comput. Graph. Stat.</em> 27 (1). Taylor &amp; Francis: 157–65. <a href="https://doi.org/10.1080/10618600.2017.1383264">https://doi.org/10.1080/10618600.2017.1383264</a>.</p>
</div>
<div id="ref-Van_Der_Maaten2014-zn">
<p>Van Der Maaten, L. 2014. “Accelerating t-SNE Using Tree-Based Algorithms.” <em>J. Mach. Learn. Res.</em> jmlr.org. <a href="http://www.jmlr.org/papers/volume15/vandermaaten14a/vandermaaten14a.pdf">http://www.jmlr.org/papers/volume15/vandermaaten14a/vandermaaten14a.pdf</a>.</p>
</div>
<div id="ref-Wattenberg2016-ji">
<p>Wattenberg, Martin, Fernanda Viégas, and Ian Johnson. 2016. “How to Use t-SNE Effectively.” <em>Distill</em> 1 (10). <a href="https://doi.org/10.23915/distill.00002">https://doi.org/10.23915/distill.00002</a>.</p>
</div>
<div id="ref-Wickham2014-jc">
<p>Wickham, Hadley. 2014. “Tidy Data.” <em>Journal of Statistical Software, Articles</em> 59 (10): 1–23. <a href="https://doi.org/10.18637/jss.v059.i10">https://doi.org/10.18637/jss.v059.i10</a>.</p>
</div>
<div id="ref-Wickham2016-gz">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Use R! Springer International Publishing. <a href="https://doi.org/10.1007/978-3-319-24277-4">https://doi.org/10.1007/978-3-319-24277-4</a>.</p>
</div>
<div id="ref-tidyverse">
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.</p>
</div>
<div id="ref-Wickham2015-cx">
<p>Wickham, Hadley, Dianne Cook, and Heike Hofmann. 2015. “Visualizing Statistical Models: Removing the Blindfold.” <em>Statistical Analysis and Data Mining: The ASA Data Science Journal</em> 8 (4). Wiley Subscription Services, Inc., A Wiley Company: 203–25. <a href="https://doi.org/10.1002/sam.11271">https://doi.org/10.1002/sam.11271</a>.</p>
</div>
<div id="ref-Wickham2011-st">
<p>Wickham, Hadley, Dianne Cook, Heike Hofmann, and Andreas Buja. 2011. “Tourr: An R Package for Exploring Multivariate Data with Projections.” <em>Journal of Statistical Software, Articles</em> 40 (2): 1–18. <a href="https://doi.org/10.18637/jss.v040.i02">https://doi.org/10.18637/jss.v040.i02</a>.</p>
</div>
<div id="ref-Wickham2017-dplyr">
<p>Wickham, Hadley, Romain Francois, Lionel Henry, and Kirill Müller. 2017. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.</p>
</div>
<div id="ref-Wilkinson2005-kq">
<p>Wilkinson, Leland. 2005. <em>The Grammar of Graphics</em>. Statistics and Computing. Springer Science &amp; Business Media. <a href="https://doi.org/10.1007/0-387-28695-0">https://doi.org/10.1007/0-387-28695-0</a>.</p>
</div>
<div id="ref-Wilkinson2005-xt">
<p>Wilkinson, Leland, Anushka Anand, and Robert Grossman. 2005. “Graph-Theoretic Scagnostics.” In <em>Proceedings of the Proceedings of the 2005 IEEE Symposium on Information Visualization</em>, 21. INFOVIS ’05. Washington, DC, USA: IEEE Computer Society. <a href="https://doi.org/10.1109/INFOVIS.2005.14">https://doi.org/10.1109/INFOVIS.2005.14</a>.</p>
</div>
<div id="ref-Xie2016-oa">
<p>Xie, Yihui. 2016. <em>Bookdown: Authoring Books and Technical Documents with R Markdown</em>. CRC Press. <a href="https://play.google.com/store/books/details?id=8nm0DQAAQBAJ">https://play.google.com/store/books/details?id=8nm0DQAAQBAJ</a>.</p>
</div>
<div id="ref-Xie2017-ke">
<p>———. 2017. <em>Dynamic Documents with R and Knitr</em>. Chapman; Hall/CRC. <a href="https://doi.org/10.1201/9781315382487">https://doi.org/10.1201/9781315382487</a>.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="6-2-further-work.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sa-lee/thesis/edit/master/Rmd//99-references.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
